# Obtain data and perform QC using Seurat{#seuratqc}

## Download data

To replicate results in this tutorial, you need to download the PBMC data from 10X website.

The PBMC multiome dataset is available from [10x genomics](https://support.10xgenomics.com/single-cell-multiome-atac-gex/datasets/1.0.0/pbmc_granulocyte_sorted_10k). The website contains output files from CelL Ranger ARC. Please go to the above website and make sure you download the following files to your data folder:

* Filtered feature barcode matrix (HDF5)
* ATAC Per fragment information file (TSV.GZ)
* ATAC Per fragment information index (TSV.GZ index)

In Terminal window:
```{bash eval=FALSE}
mkdir pbmc_data
cd pbmc_data
# download data
wget https://cf.10xgenomics.com/samples/cell-arc/1.0.0/pbmc_granulocyte_sorted_10k/pbmc_granulocyte_sorted_10k_filtered_feature_bc_matrix.h5
wget https://cf.10xgenomics.com/samples/cell-arc/1.0.0/pbmc_granulocyte_sorted_10k/pbmc_granulocyte_sorted_10k_atac_fragments.tsv.gz
wget https://cf.10xgenomics.com/samples/cell-arc/1.0.0/pbmc_granulocyte_sorted_10k/pbmc_granulocyte_sorted_10k_atac_fragments.tsv.gz.tbi

# check files
ls
```

## Load data in R and construct Seurat object

For the following processing and analysis steps, we are going to use `Seurat` and its sister package `Signac`. `Seurat` is a popular package in R that provides users with well-curated functions and workflows. Seurat was first developed for the clustering of scRNA-seq data, but with continuing updates in the last few years, this package has become a popular tool for QC, analysis, and exploration of scRNA-seq data as well. For more information on `Seurat`, see the [Seurat website](https://satijalab.org/seurat/) from the Satija Lab, which has very nice documentation and links to the Satija Lab [publications](https://satijalab.org/publications) as well as detailed [tutorials and vignettes](https://satijalab.org/seurat/vignettes.html).

In this notebook, we will use the most recent version of Seurat, version 5.

We first load all required libraries in this tutorial.
```{r eval=TRUE,results=FALSE,warning=FALSE,message=FALSE}
library(Seurat)
library(SeuratData)
library(Signac)
library(BSgenome.Hsapiens.UCSC.hg38) 
library(EnsDb.Hsapiens.v86)
library(ggplot2)
library(cowplot)
library(hdf5r)
library(dplyr)
library(patchwork)
library(clustree)
library(TFBSTools)
library(JASPAR2020)
library(BiocParallel)
library(motifmatchr)
```

Also set working directory to the folder where you saved your downloaded datasets.
```{r eval=FALSE}
# check current working directory
getwd()

# set working directory
setwd("/dfs6/pub/your_user_name/pbmc_data")
getwd() # check again
```


Next, we will read in the `.h5` file into R and construct Seurat Object with both scRNA-seq and scATAC-seq assays.
```{r eval=F,results='markup',warning=FALSE,message=FALSE}
# the 10x hdf5 file contains both data types. 
inputdata.10x <- Read10X_h5("pbmc_granulocyte_sorted_10k_filtered_feature_bc_matrix.h5")
# extract RNA and ATAC data
rna_counts <- inputdata.10x$`Gene Expression`
atac_counts <- inputdata.10x$Peaks

## if reading files from "filtered_feature_bc_matrix"
# inputdata.10x <- Read10X(data.dir = "filtered_feature_bc_matrix")

```

```{r eval=F,results='markup',warning=FALSE,message=FALSE}
# Create Seurat object
pbmc <- CreateSeuratObject(counts = rna_counts,project="pbmc")

# Now add in the ATAC-seq data
# we'll only use peaks in standard chromosomes
grange.counts <- StringToGRanges(rownames(atac_counts), sep = c(":", "-"))
grange.use <- seqnames(grange.counts) %in% standardChromosomes(grange.counts)
atac_counts <- atac_counts[as.vector(grange.use), ]
annotations <- GetGRangesFromEnsDb(ensdb = EnsDb.Hsapiens.v86)
seqlevelsStyle(annotations) <- 'UCSC'
genome(annotations) <- "hg38"

frag.file <- "pbmc_granulocyte_sorted_10k_atac_fragments.tsv.gz"
chrom_assay <- CreateChromatinAssay(
   counts = atac_counts,
   sep = c(":", "-"),
   genome = 'hg38',
   fragments = frag.file,
   min.cells = 10,
   annotation = annotations
 )
pbmc[["ATAC"]] <- chrom_assay

# check data
pbmc
```

Add something about seurat object structure
Explore Seurat Objects using `@` or `$`
```{r eval=FALSE}
# check available assays in data
pbmc@active.assay

#check a single assay
pbmc[["RNA"]]
pbmc[["ATAC"]]

# check elements within each assay

pbmc[["RNA"]]$counts

# check metadata
pbmc@meta.data
```

Saving data periodically to preserve Seurat results. Output data can be large.

```{r eval=F, echo=F}
saveRDS(pbmc,file="pbmc.rds")
```

## Perform quality control using both assays
```{r eval=T,echo=F}
pbmc<-readRDS("../pbmc_data/pbmc.rds")
```

Using `DefaultAssay()` to specify the assay you want to work on. `PercentageFeatureSet` calculates the proportion of transcripts mapping to mitochondrial genes. Large amount of mitochondrial contamination is an indicator of dead or dying cells. Voilin plots help to visualize the distribution of each parameter.

```{r eval=FALSE,warning=FALSE,message=FALSE}
# calculate percentage of mitochondrial genes in each cell
DefaultAssay(pbmc) <- "RNA"
pbmc[["percent.mt"]] <- PercentageFeatureSet(pbmc, pattern = "^MT-")
DefaultAssay(pbmc) <- "ATAC"
pbmc <- NucleosomeSignal(pbmc)
pbmc <- TSSEnrichment(pbmc)

DensityScatter(pbmc, x = 'nCount_ATAC', y = 'TSS.enrichment', log_x = TRUE, quantiles = TRUE)

# generate violin plot
VlnPlot(pbmc, features = c( "nCount_RNA", "nFeature_RNA", "percent.mt"), ncol = 3,
  log = TRUE, pt.size = 0) + NoLegend()
VlnPlot(pbmc, features = c("nCount_ATAC", "TSS.enrichment", "nucleosome_signal"), ncol = 3,
  log = TRUE, pt.size = 0) + NoLegend()
# change pt.size to show scatter p
```

```{r eval=T,fig.cap='Voilin plot for QC', out.width='100%', fig.asp=.75, fig.align='center'}
# show plot
VlnPlot(pbmc, features = c( "nCount_RNA", "nFeature_RNA", "percent.mt"), ncol = 3,
  log = TRUE, pt.size = 0) + NoLegend()
VlnPlot(pbmc, features = c("nCount_ATAC", "TSS.enrichment", "nucleosome_signal"), ncol = 3,
  log = TRUE, pt.size = 0) + NoLegend()
```

```{r eval=T,fig.cap='Voilin plot for QC', out.width='100%', fig.asp=.75, fig.align='center'}
# Scatter plot for nCount_RNA, percent.mito, nFeature_RNA and filtering based on these variables
# FeatureScatter is typically used to visualize feature-feature relationships, but can be used
# for anything calculated by the object, i.e. columns in object metadata, PC scores etc.

plot1 <- FeatureScatter(pbmc, feature1 = "nCount_RNA", feature2 = "percent.mt")
plot2 <- FeatureScatter(pbmc, feature1 = "nCount_RNA", feature2 = "nFeature_RNA")
library(patchwork)
plot1 + plot2 +plot_layout(ncol=2)

```


Perform QC on cells based on the parameters, you can change the paramters in the filtering criteria and see how this can change the number fo cells in the filtered object. **Note that you need to save the filtered object into a name different from the original raw data to avoid overwriting.**

```{r eval=FALSE,warning=FALSE,message=FALSE}
# play with different filtering cutoffs to see how number of cells change
pbmc_qc <- subset(
  x = pbmc,
  subset = nCount_ATAC < 7e4 &
    nCount_ATAC > 5e3 &
    nCount_RNA < 25000 &
    nCount_RNA > 1000 &
    percent.mt < 25 &
    nucleosome_signal < 2 &
    TSS.enrichment > 1
)

pbmc_qc

```

Saving data periodically to preserve Seurat results. Output data can be large.

```{r eval=F}
saveRDS(pbmc_qc,file="pbmc_qc.rds")
```
*******
