# Dimension reduction and clustering

```{r eval=TRUE,echo =F,results=FALSE,warning=FALSE,message=FALSE}
library(Seurat)
library(SeuratData)
library(Signac)
library(SeuratDisk)
library(BSgenome.Hsapiens.UCSC.hg38) 
library(EnsDb.Hsapiens.v86)
library(ggplot2)
library(cowplot)
library(hdf5r)
library(dplyr)
library(patchwork)
library(clustree)
```


```{r eval=T, echo=F}
pbmc<-readRDS("../pbmc_data/pbmc_norm.rds")
```

## Identify top variable features

```{r eval=T,results=F,warning=FALSE,message=FALSE}
# Finding highly variable genes (feature selection), we return 2000 features per dataset by default
pbmc <- FindVariableFeatures(object = pbmc, selection.method ="vst", 
                                 mean.cutoff = c(0.1,6), dispersion.cutoff = c(0.5,Inf),verbose=TRUE)
length(VariableFeatures(pbmc))
# [1] 2000 # return top 2000 variable features by default
```

```{r eval=T,echo=F, fig.cap='Label the top 10 variable features', out.width='100%', fig.asp=.75, fig.align='center'}
library(patchwork)
# Identify the 10 most highly variable genes
top10 <- head(VariableFeatures(pbmc), 10)
top10 # check top 10 highly variable genes
# [1] "JCHAIN"  "PTGDS"   "IGKC"    "IGHA1"   "TXNDC5"  "TCF4"    "SOX5"    "MZB1"    "ZNF385D" "CDKN1C" 

# plot variable features with and without labels
plot1 <- VariableFeaturePlot(pbmc)
plot2 <- LabelPoints(plot = plot1, points = top10, repel = TRUE)
plot1 + plot2 + plot_layout(ncol=1)
```

## Dimension reduction
Run separately for RNA and ATAC. `min.dist` and `spread` controls how tightly the points are clustered.

```{r eval=T,warning=FALSE,message=FALSE}
# RNA analysis
DefaultAssay(pbmc) <- "RNA"
pbmc <- RunPCA(pbmc) 
DimPlot(object = pbmc)
# determin if cell cycle has major variation
# Running a PCA on cell cycle genes reveals
DimPlot(pbmc,reduction = "pca",group.by= "Phase",split.by = "Phase")
```

Other visualization figures for determining dimensionality:

```{r eval=T, fig.cap='Variations in top genes for the first two PCs', out.width='100%', fig.asp=.75, fig.align='center'}
# visualize the top genes associated with reduction components for the first 2 PCs
VizDimLoadings(object = pbmc, dims = 1:2)
```

```{r eval=T,fig.cap='Heatmap for visualizing heterogeneity in PCs', out.width='100%', fig.asp=.75, fig.align='center'}
# use heatmap
DimHeatmap(object = pbmc, dims = 1:9, cells = 500, balanced = TRUE)
```

```{r eval=T,fig.cap='Checking elbow plot is the most common way for determining dimensionality', out.width='100%', fig.asp=.75, fig.align='center'}
ElbowPlot(object = pbmc)
```

We will do dimension reduction for scATAC-seq as well.

```{r eval=F,warning=FALSE,message=FALSE}
# ATAC analysis
DefaultAssay(pbmc) <- "ATAC"
pbmc <- RunTFIDF(pbmc)
pbmc <- FindTopFeatures(pbmc, min.cutoff = 'q0')
pbmc <- RunSVD(pbmc)
pbmc <- RunUMAP(pbmc, reduction = 'lsi', dims = 2:50, reduction.name = "umap.atac", reduction.key = "atacUMAP_")
```

## Clustering and joint clustering

```{r eval=T, echo=F}
pbmc<-readRDS("../pbmc_data/pbmc_anno.rds")
```

Clustering based on RNA assay. Play around with different resolutions.
```{r eval=F,warning=FALSE,message=FALSE}
DefaultAssay(pbmc) <- "RNA"
pbmc <- FindNeighbors(pbmc, dims = 1:50,force.recalc = T)
pbmc <- FindClusters(pbmc, algorithm = 3, resolution = 0.7, group.singletons=T)
pbmc<-RunUMAP(object = pbmc, dims = 1:50, reduction.name = "umap.rna", reduction.key = 'rnaUMAP_',
                    min.dist=0.3, spread=0.5,assay="SCT",slot="scale.data")
```

```{r eval=T,fig.cap='Resolution=0.7', out.width='100%', fig.asp=.75, fig.align='center'}
DimPlot(pbmc, label = TRUE, repel = TRUE, reduction = "umap.rna")
```

Generate a resolution tree for res=0.1 to 1.5 to select a suitable resolution.
```{r eval=T, result=FALSE, warning=FALSE,message=FALSE}
DefaultAssay(pbmc) <- "RNA"
resolutions <- seq(0.1, 1.5, by = 0.1)
# Initialize a list to store the clustering results
clustering.list <- list()
# Loop over the resolution values and perform clustering
for (i in 1:length(resolutions)){
	# Run the FindClusters function with the resolution parameter
	pbmc <- FindClusters(pbmc, algorithm = 3, resolution = resolutions[i], group.singletons=T, verbose = TRUE)
	# Store the cluster labels in the list
	clustering.list[[i]] <- Idents(pbmc) 
	print(i)
}
names(clustering.list)<-paste0("res_",resolutions)

```

```{r eval=T,fig.cap='Clusters based on RNA assay vs joint assays', out.width='100%', fig.asp=1.2, fig.align='center'}
# Load the clustree package
library(clustree)
df<- as.data.frame(lapply(clustering.list,as.factor))
# Plot the cluster tree for different resolution values
clustree(df, prefix = "res_")
```

Compare: resolution=0.1, resolution=0.6 and resolution=1.1.
s
```{r eval=T,fig.cap='Clusters based on RNA assay vs joint assays', out.width='250%', fig.asp=.4, fig.align='center'}
p1 <- DimPlot(pbmc, reduction = "wnn.umap", group.by = "RNA_snn_res.0.1", label = TRUE, label.size = 2.5, repel = TRUE) + ggtitle("RNA clusters res=0.1")
p2 <- DimPlot(pbmc, reduction = "wnn.umap", group.by = "RNA_snn_res.0.7", label = TRUE, label.size = 2.5, repel = TRUE) + ggtitle("RNA clusters res=0.7")
p3 <- DimPlot(pbmc, reduction = "wnn.umap", group.by = "RNA_snn_res.1.1", label = TRUE, label.size = 2.5, repel = TRUE) + ggtitle("RNA clusters res=1.1")
p1 + p2 + p3 & NoLegend() & theme(plot.title = element_text(hjust = 0.5))

```

```{r eval=F,warning=FALSE,message=FALSE}
# We exclude the first dimension for ATAC as this is typically correlated with sequencing depth
pbmc <- FindMultiModalNeighbors(pbmc, reduction.list = list("pca", "lsi"), dims.list = list(1:50, 2:50))
pbmc <- RunUMAP(pbmc, nn.name = "weighted.nn", reduction.name = "wnn.umap", reduction.key = "wnnUMAP_",min.dist=0.3, spread=0.5)
pbmc <- FindClusters(pbmc, graph.name = "wsnn", algorithm = 3, resolution = 0.6, group.singletons=T,verbose = FALSE)
```

You can further generate sub-clusters within large cluster to explore underlying structure

```{r eval=F,warning=FALSE,message=FALSE}
# pbmc <- FindSubCluster(pbmc, cluster = 6, graph.name = "wsnn", algorithm = 3)
# Idents(pbmc) <- "sub.cluster"
```

Compare: resolution=0.7 for RNA and resolution=0.6 for joint assays have similar number of clusters.
s
```{r eval=T,fig.cap='Clusters based on RNA assay vs joint assays', out.width='150%', fig.asp=.5, fig.align='center'}
p1 <- DimPlot(pbmc, reduction = "wnn.umap", group.by = "RNA_snn_res.0.7", label = TRUE, label.size = 2.5, repel = TRUE) + ggtitle("RNA clusters")
p2 <- DimPlot(pbmc, reduction = "wnn.umap", group.by = "wsnn_res.0.6", label = TRUE, label.size = 2.5, repel = TRUE) + ggtitle("WNN clusters")
p1 + p2 & NoLegend() & theme(plot.title = element_text(hjust = 0.5))

```

compare: same cluster distribution on different assay space.

```{r eval=T,fig.cap='Clusters for single assay UMAP space vs joint assay space', out.width='250%', fig.asp=.4, fig.align='center'}
p1 <- DimPlot(pbmc, reduction = "umap.rna", group.by = "seurat_clusters", label = TRUE, label.size = 2.5, repel = TRUE) + ggtitle("RNA")
p2 <- DimPlot(pbmc, reduction = "umap.atac", group.by = "seurat_clusters", label = TRUE, label.size = 2.5, repel = TRUE) + ggtitle("ATAC")
p3 <- DimPlot(pbmc, reduction = "wnn.umap", group.by = "seurat_clusters", label = TRUE, label.size = 2.5, repel = TRUE) + ggtitle("WNN")
p1 + p2 + p3 & NoLegend() & theme(plot.title = element_text(hjust = 0.5))

```


```{r eval=F}
saveRDS(pbmc,file="pbmc_clust.rds")
```


