--- 
title: "Hands-on Analysis on Single Cell Multiome"
author: "Danni Liu"
date: "`r Sys.Date()`"
site: bookdown::bookdown_site
output: bookdown::gitbook
documentclass: book
bibliography: [book.bib, packages.bib]
biblio-style: apalike
link-citations: yes
github-repo: rstudio/bookdown-demo
description: "This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook."
---

# Prerequisites

Placeholder


## About Data
## Server environment requirements
## R libraries
## Construct folder structure
## Downloading FASTQ files

<!--chapter:end:index.Rmd-->


# Cell Ranger ARC for preprocessing sequencing data {#cellranger_arc}

Placeholder


## `cellranger-arc mkfastq`
## BCL naming convention
## `cellranger-arc count`

<!--chapter:end:01-CellRangerARC.Rmd-->


# Obtain data and perform QC using Seurat{#seuratqc}

Placeholder


## Download data
## Load data in R and construct Seurat object
## Perform quality control using both assays

<!--chapter:end:02-SeuratQC.Rmd-->


# Normalization and cell cycle correction

Placeholder


## Normalization
## Cell cycle correction and scale data

<!--chapter:end:03-norm_cc.Rmd-->


# Dimension reduction and clustering

Placeholder


## Identify top variable features
## Dimension reduction
## Clustering and joint clustering

<!--chapter:end:04-cluster.Rmd-->

# Differential Expression (DE) analysis and cell type annotation

```{r eval=TRUE,echo =F,results=FALSE,warning=FALSE,message=FALSE}
library(Seurat)
library(SeuratData)
library(Signac)
library(SeuratDisk)
library(BSgenome.Hsapiens.UCSC.hg38) 
library(EnsDb.Hsapiens.v86)
library(ggplot2)
library(cowplot)
library(hdf5r)
library(dplyr)
library(patchwork)
library(clustree)
```

Continue from checkpoint...

```{r eval=T}
pbmc<-readRDS("../pbmc_data/pbmc_clust.rds")
```

## DE analysis

```{r eval=T,warning=FALSE,message=FALSE}
DefaultAssay(pbmc) <- "RNA"
# cluster 0 vs all others
# this can take several minutes
cluster0.markers <- FindMarkers(object = pbmc, ident.1 = 0, min.pct = 0.1)
print(x = head(x = cluster0.markers, n = 5))

```

In the code below, we can also use FindAllMarkers function to automate this process for all clusters. This compares 0 with all other clusters, 1 with all other clusters, etc. Note here that we have specified that min.pct =0.25 , as we want the log(fold-change) threshold for a gene to be at least 0.25 to be identified as significant. Then we select the top 15 markers for each cluster, based on log(fold-changes, but it will take long time to finish.

```{r eval=F}
DefaultAssay(pbmc) <- "RNA"
# find markers for every cluster compared to all remaining cells, report only the positive ones
# FindAllMarkers can take long time to run
allmarkers <-FindAllMarkers(pbmc, only.pos = FALSE, min.pct = 0.25, logfc.threshold = 0.25,assay="RNA",slot="scale.data")

library(dplyr)
allmarkers_grouped<- allmarkers %>%
    group_by(cluster) %>% slice_max(n = 1, order_by = avg_log2FC)
write.table(as.data.frame(allmarkers),file="../pbmc_data/DE_markers.txt",sep="\t",row.names = F,col.names = T)

```

Visualizing DE markers.

```{r eval=T,warning=FALSE,message=FALSE,fig.cap='Violin plot for gene TSHZ2 and LEF1', out.width='100%', fig.asp=.5, fig.align='center'}
allmarkers<-read.table("../pbmc_data/DE_markers.txt",header=T)
top3_log2FC <- allmarkers %>% group_by(cluster) %>% top_n(3, avg_log2FC)
head(top3_log2FC)
# plot
features<-unique(top3_log2FC$gene)
VlnPlot(pbmc, features = c("TSHZ2","LEF1"),slot = "counts", log = TRUE, pt.size=0)
```
```{r eval=T, fig.cap='Feature expression plot', out.width='100%', fig.asp=1, fig.align='center'}
FeaturePlot(pbmc, features = c("LEF1", "FHIT", "LTB", "NELL2", "AFF3", "BANK1", "VCAN", "FCGR3A", "TCF4"))
```

```{r eval=T, fig.cap='Heatmap of top 3 DE markers for each cluster', out.width='300%', fig.asp=.75, fig.align='center'}
DoHeatmap(object = pbmc, features = top3_log2FC$gene)+ NoLegend()
```

## Cell type annotation

We annotate the clusters below. Note that you could also annotate the dataset using supervised mapping pipelines or automated web tool, [Azimuth](https://satijalab.org/azimuth/).

To annotate cell types in the dataset we can transfer cell labels from an existing PBMC reference dataset using tools in the Seurat package. See the Seurat reference mapping vignette for more information.

Weâ€™ll use an annotated PBMC reference dataset from Hao et al. (2020), available for download here: https://atlas.fredhutch.org/data/nygc/multimodal/pbmc_multimodal.h5seurat

Note that the SeuratDisk package is required to load the reference dataset. Installation instructions for SeuratDisk can be found here.


```{bash eval=F}
# download reference annotation
wget https://atlas.fredhutch.org/data/nygc/multimodal/pbmc_multimodal.h5seurat
```

```{r eval=F,warning=FALSE,message=FALSE}
library(SeuratDisk)
# load PBMC reference
reference <- LoadH5Seurat("../pbmc_data/pbmc_multimodal.h5seurat")
DefaultAssay(pbmc) <- "SCT"

# transfer cell type labels from reference to query
transfer_anchors <- FindTransferAnchors(
  reference = reference, query = pbmc, 
	reference.reduction="spca", normalization.method = "SCT", 
	dims = 1:50 )
```

MapQuery() is a wrapper around three functions: TransferData(), IntegrateEmbeddings(), and ProjectUMAP(). TransferData() is used to transfer cell type labels and impute the ADT values. IntegrateEmbeddings() and ProjectUMAP() are used to project the query data onto the UMAP structure of the reference. 

```{r, eval=F}
# MapQuery() is a wrapper around three functions: TransferData(), IntegrateEmbeddings(), and ProjectUMAP()
pbmc <- MapQuery(
  anchorset = transfer_anchors,
  query = pbmc,
  reference = reference,
  refdata = list(
    celltype = "celltype.l2",
    predicted_ADT = "ADT"
  ),
  reference.reduction = "spca", # reference has different types of reduction performed, pca, spca, umap ,etc.
  reduction.model = "wnn.umap",
	transferdata.args=list(),
	integrateembeddings.args = list(),
  projectumap.args = list()
)

```


```{r eval=T, echo=F}
pbmc<-readRDS("../pbmc_data/pbmc_anno.rds")
```

```{r eval=T,fig.cap='Clusters for annotated cell types', out.width='150%', fig.asp=.6, fig.align='center'}
# plot on reference data space
p1<-DimPlot(pbmc, reduction = "ref.umap", group.by = "predicted.celltype.l2", label = TRUE, label.size = 3 ,repel = TRUE)
# plot on wnn space
p2<-DimPlot(pbmc, reduction = "wnn.umap", group.by = "predicted.celltype.l2", label = TRUE, label.size = 3 ,repel = TRUE)
p1 + p2 & NoLegend() & theme(plot.title = element_text(hjust = 0.5))
```

```{r eval=F}
saveRDS(pbmc,file="pbmc_anno.rds")
```

<!--chapter:end:05-DEanalysis.Rmd-->


# Peak calling and Motif analysis

Placeholder


## Linkage plot
## Call peaks using MACS2
## Motif analysis

<!--chapter:end:06-motif.Rmd-->


# Data integration

Placeholder



<!--chapter:end:07-integration.Rmd-->

`r if (knitr:::is_html_output()) '
# References {-}

'`

<!--chapter:end:08-references.Rmd-->

