--- 
title: "Hands-on Analysis on Single Cell Multiome"
author: "Danni Liu"
date: "`r Sys.Date()`"
site: bookdown::bookdown_site
output: bookdown::gitbook
documentclass: book
bibliography: [book.bib, packages.bib]
biblio-style: apalike
link-citations: yes
github-repo: rstudio/bookdown-demo
description: "This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook."
---

# Prerequisites

## About Data

The data used in this hands-on tutorial is a public single cell 'multiome' dataset of cryopreserved human peripheral blood mononuclear cells (PBMC) from 10x Genomics. This dataset is collected from a 25-year-old healthy female donor with around 10,000 cells, and scRNA-seq and scATAC-seq profiles were simultaneously sequenced from the same cells. You can find all information about this data and Cell Ranger ARC output reports from [10x Genomics website](https://support.10xgenomics.com/single-cell-multiome-atac-gex/datasets/1.0.0/pbmc_granulocyte_sorted_10k?). 

In this hands-on tutorial, we present a complete workflow for analyzing multiome data starting from sequencing data, with an additional section for data integration of multiomics datasets that are collected from different experiments. We will demonstrate how to preprocess raw sequencing data using Cell Ranger ARC, perform data quality control, cluster and annotate cell types, and identify marker genes using differential expression (DE) analysis. Then we will treat the two profiles of scRNA-seq and scATAC-seq as originating from two separate experiments, and perform data integration. Since they were originally measured in the same cells, this provides a ground truth that we can use to assess the accuracy of integration.

::: {#note style="color: red;"}
The PBMC dataset is used here for illustration purpose. For hands-on practice, please try to apply the pipeline in this tutorial on the mice pancreatic data. This can be an experience that highly assembles the real-world data analysis process, where we apply an existing pipeline from a publication to a new target dataset and make modifications of the pipeline to fit for our purpose. 
:::


## Server environment requirements
We will use the UCI HPC3 environment for analysis. Please navigate to the [JypyterHub portal](https://hpc3.rcic.uci.edu/biojhub4/hub/login), and start a container under the one with ** *CaSB Short Course ** which has "R4.3.3" and required R packages installed in the environment. Please request **4-8 CPU cores**, with **6Gb of memory** for each core. Also please check the last box "Enable full HPC3 software stack (overide container stack)" if you want to try running `cellranger-arc`.

```{r echo=FALSE, fig.align = 'center', out.width = "100%", fig.cap = "JupyterHub server setting", }
knitr::include_graphics("images/hpc3.png")
```

Once the server started, click to open "Rstudio" app or a "Notebook" app under R kernel.

## R libraries

If you want to run analysis using your own laptop, please make sure you have the latest R and RStudio installed. You can refer to [RStudio website](https://posit.co/download/rstudio-desktop/) for installation instructions. 

Below are the codes needed to install the necessary R packages if you are using your laptop. `Seurat` and `Signac` are the major packages for analysis in this tutorial.

```{r eval=FALSE}
# Enter commands in R (or R studio, if installed)
install.packages('Seurat')
library(Seurat)

#install.packages("devtools")
#library(devtools)
#devtools::install_github('satijalab/seurat-data')

setRepositories(ind=1:3) # needed to automatically install Bioconductor dependencies
install.packages("Signac")

if (!requireNamespace("remotes", quietly = TRUE)) {
  install.packages("remotes")
}
remotes::install_github("mojaveazure/seurat-disk")

# Installing genome assembly and gene annotation packages
if (!require("BiocManager", quietly = TRUE))
    install.packages("BiocManager")
BiocManager::install(c('BSgenome.Hsapiens.UCSC.hg38', 'EnsDb.Hsapiens.v86'))

# install other packages needed.
install.packages("hdf5r")
BiocManager::install("biovizBase")
BiocManager::install("glmGamPoi")
install.packages("dplyr")
install.packages("clustree")
BiocManager::install("TFBSTools")
BiocManager::install("JASPAR2020")
BiocManager::install("BiocParallel")
BiocManager::install("motifmatchr")
install.packages('ggseqlogo')
# install.packages("VariantAnnotation_1.50.0.tgz", repos = NULL, type="source")
```

* Seurat installation: https://satijalab.org/seurat/articles/install_v5
* Signac installation: https://stuartlab.org/signac/articles/install
* SeuratData installation: https://github.com/satijalab/seurat-data

## Construct folder structure in Terminal

Some basic bash/shell commands
```{bash eval=F}
# check where you are at
pwd
# create directory
mkdir new_folder
# list all files in directory
ls
ls new_folder
# copy file
cp file newfolder/file
#remove file--can't recover once removed
rm file
# count number of lines
wc -l file
# count number of columns
head -1 file | awk '{print NF}'

```

Navigate to the working directory, with 1TB of storage. 

```{r eval=FALSE}
cd /dfs6/pub/your_username
pwd # check the current directory path
ls # list all files in current directory

```

## Accessing pancreas data

The sequencing data was processed using `cellranger-arc` so you will directly load the output file into R. The data was saved in `/dfs8/commondata/workshop/CaSB_data`. You can copy the data to your working directory. 

```{bash eval=F}
scp -r /dfs8/commondata/workshop/CaSB_data /dfs6/pub/your_username/.
cd /dfs6/pub/your_username/CaSB_data
ls
```


```{r include=FALSE}
# automatically create a bib database for R packages
knitr::write_bib(c(
  .packages(), 'bookdown', 'knitr', 'rmarkdown'
), 'packages.bib')
```
