--- 
title: "Hands-on Analysis on Single Cell Multiome"
author: "Danni Liu"
date: "`r Sys.Date()`"
site: bookdown::bookdown_site
output: bookdown::gitbook
documentclass: book
bibliography: [book.bib, packages.bib]
biblio-style: apalike
link-citations: yes
github-repo: rstudio/bookdown-demo
description: "This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook."
---

# Environment requirements

Need to install the following packages before running the codes.
```{r eval=FALSE}
# Enter commands in R (or R studio, if installed)
r = getOption("repos")
r["CRAN"] = "http://cran.us.r-project.org"
options(repos = r)

install.packages('Seurat')
library(Seurat)

install.packages("devtools")
library(devtools)
devtools::install_github('satijalab/seurat-data')

setRepositories(ind=1:3) # needed to automatically install Bioconductor dependencies
install.packages("Signac")
```

- Seurat installation: https://satijalab.org/seurat/articles/install_v5
- Signac installation: https://stuartlab.org/signac/articles/install
- SeuratData installation: https://github.com/satijalab/seurat-data

# About Data

The data used in this hands-on tutorial is a public single cell 'multiome' dataset of cryopreserved human peripheral blood mononuclear cells (PBMC) from 10x Genomics. This dataset is collected from a 25-year-old healthy female donor with around 10,000 cells, and scRNA-seq and scATAC-seq profiles were simultaneously sequenced from the same cells. You can find all information about this data and Cell Ranger ARC output reports from [10x Genomics website](https://support.10xgenomics.com/single-cell-multiome-atac-gex/datasets/1.0.0/pbmc_granulocyte_sorted_10k?). 

In this hands-on tutorial, we present a complete workflow for analyzing multiome data, with an additional section for data integration of multiomics. We will demonstrate how to preprocess raw sequencing data using Cell Ranger, perform data quality control, clustering and cell type annotation, and identify marker genes using differential expression (DE) analysis. Then we will treat the two profiles of scRNA-seq and scATAC-seq as originating from two separate experiments, and perform data integration. Since they were originally measured in the same cells, this provides a ground truth that we can use to assess the accuracy of integration.


This is a _sample_ book written in **Markdown**. You can use anything that Pandoc's Markdown supports, e.g., a math equation $a^2 + b^2 = c^2$.

The **bookdown** package can be installed from CRAN or Github:


```{r include=FALSE}
# automatically create a bib database for R packages
knitr::write_bib(c(
  .packages(), 'bookdown', 'knitr', 'rmarkdown'
), 'packages.bib')
```
