<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 8 Data integration | Hands-on Analysis on Single Cell Multiome</title>
  <meta name="description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  <meta name="generator" content="bookdown 0.39 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 8 Data integration | Hands-on Analysis on Single Cell Multiome" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 8 Data integration | Hands-on Analysis on Single Cell Multiome" />
  
  <meta name="twitter:description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  

<meta name="author" content="Danni Liu" />


<meta name="date" content="2024-06-07" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="peak-calling-and-motif-analysis.html"/>
<link rel="next" href="references.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Hands-on Analysis on Single Cell Multiome</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Prerequisites</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#about-data"><i class="fa fa-check"></i><b>1.1</b> About Data</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#server-environment-requirements"><i class="fa fa-check"></i><b>1.2</b> Server environment requirements</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#r-libraries"><i class="fa fa-check"></i><b>1.3</b> R libraries</a></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#construct-folder-structure-in-terminal"><i class="fa fa-check"></i><b>1.4</b> Construct folder structure in Terminal</a></li>
<li class="chapter" data-level="1.5" data-path="index.html"><a href="index.html#accessing-pancreas-data"><i class="fa fa-check"></i><b>1.5</b> Accessing pancreas data</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="cellranger_arc.html"><a href="cellranger_arc.html"><i class="fa fa-check"></i><b>2</b> Cell Ranger ARC for preprocessing sequencing data</a>
<ul>
<li class="chapter" data-level="2.1" data-path="cellranger_arc.html"><a href="cellranger_arc.html#cellranger-arc-mkfastq"><i class="fa fa-check"></i><b>2.1</b> <code>cellranger-arc mkfastq</code></a></li>
<li class="chapter" data-level="2.2" data-path="cellranger_arc.html"><a href="cellranger_arc.html#bcl-naming-convention"><i class="fa fa-check"></i><b>2.2</b> BCL naming convention</a></li>
<li class="chapter" data-level="2.3" data-path="cellranger_arc.html"><a href="cellranger_arc.html#cellranger-arc-count"><i class="fa fa-check"></i><b>2.3</b> <code>cellranger-arc count</code></a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="seuratqc.html"><a href="seuratqc.html"><i class="fa fa-check"></i><b>3</b> Obtain data and perform QC using Seurat</a>
<ul>
<li class="chapter" data-level="3.1" data-path="seuratqc.html"><a href="seuratqc.html#download-data"><i class="fa fa-check"></i><b>3.1</b> Download data</a></li>
<li class="chapter" data-level="3.2" data-path="seuratqc.html"><a href="seuratqc.html#load-data-in-r-and-construct-seurat-object"><i class="fa fa-check"></i><b>3.2</b> Load data in R and construct Seurat object</a></li>
<li class="chapter" data-level="3.3" data-path="seuratqc.html"><a href="seuratqc.html#perform-quality-control-using-both-assays"><i class="fa fa-check"></i><b>3.3</b> Perform quality control using both assays</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="normalization-and-cell-cycle-correction.html"><a href="normalization-and-cell-cycle-correction.html"><i class="fa fa-check"></i><b>4</b> Normalization and cell cycle correction</a>
<ul>
<li class="chapter" data-level="4.1" data-path="normalization-and-cell-cycle-correction.html"><a href="normalization-and-cell-cycle-correction.html#normalization"><i class="fa fa-check"></i><b>4.1</b> Normalization</a></li>
<li class="chapter" data-level="4.2" data-path="normalization-and-cell-cycle-correction.html"><a href="normalization-and-cell-cycle-correction.html#cell-cycle-correction-and-scale-data"><i class="fa fa-check"></i><b>4.2</b> Cell cycle correction and scale data</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="dimension-reduction-and-clustering.html"><a href="dimension-reduction-and-clustering.html"><i class="fa fa-check"></i><b>5</b> Dimension reduction and clustering</a>
<ul>
<li class="chapter" data-level="5.1" data-path="dimension-reduction-and-clustering.html"><a href="dimension-reduction-and-clustering.html#identify-top-variable-features"><i class="fa fa-check"></i><b>5.1</b> Identify top variable features</a></li>
<li class="chapter" data-level="5.2" data-path="dimension-reduction-and-clustering.html"><a href="dimension-reduction-and-clustering.html#dimension-reduction"><i class="fa fa-check"></i><b>5.2</b> Dimension reduction</a></li>
<li class="chapter" data-level="5.3" data-path="dimension-reduction-and-clustering.html"><a href="dimension-reduction-and-clustering.html#clustering-and-joint-clustering"><i class="fa fa-check"></i><b>5.3</b> Clustering and joint clustering</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="differential-expression-de-analysis-and-cell-type-annotation.html"><a href="differential-expression-de-analysis-and-cell-type-annotation.html"><i class="fa fa-check"></i><b>6</b> Differential Expression (DE) analysis and cell type annotation</a>
<ul>
<li class="chapter" data-level="6.1" data-path="differential-expression-de-analysis-and-cell-type-annotation.html"><a href="differential-expression-de-analysis-and-cell-type-annotation.html#de-analysis"><i class="fa fa-check"></i><b>6.1</b> DE analysis</a></li>
<li class="chapter" data-level="6.2" data-path="differential-expression-de-analysis-and-cell-type-annotation.html"><a href="differential-expression-de-analysis-and-cell-type-annotation.html#cell-type-annotation"><i class="fa fa-check"></i><b>6.2</b> Cell type annotation</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="peak-calling-and-motif-analysis.html"><a href="peak-calling-and-motif-analysis.html"><i class="fa fa-check"></i><b>7</b> Peak calling and Motif analysis</a>
<ul>
<li class="chapter" data-level="7.1" data-path="peak-calling-and-motif-analysis.html"><a href="peak-calling-and-motif-analysis.html#linkage-plot"><i class="fa fa-check"></i><b>7.1</b> Linkage plot</a></li>
<li class="chapter" data-level="7.2" data-path="peak-calling-and-motif-analysis.html"><a href="peak-calling-and-motif-analysis.html#call-peaks-using-macs2"><i class="fa fa-check"></i><b>7.2</b> Call peaks using MACS2</a></li>
<li class="chapter" data-level="7.3" data-path="peak-calling-and-motif-analysis.html"><a href="peak-calling-and-motif-analysis.html#motif-analysis"><i class="fa fa-check"></i><b>7.3</b> Motif analysis</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="data-integration.html"><a href="data-integration.html"><i class="fa fa-check"></i><b>8</b> Data integration</a>
<ul>
<li class="chapter" data-level="8.1" data-path="data-integration.html"><a href="data-integration.html#identifying-anchors-between-scrna-seq-and-scatac-seq-datasets"><i class="fa fa-check"></i><b>8.1</b> Identifying anchors between scRNA-seq and scATAC-seq datasets</a></li>
<li class="chapter" data-level="8.2" data-path="data-integration.html"><a href="data-integration.html#annotate-scatac-seq-cells-via-label-transfer"><i class="fa fa-check"></i><b>8.2</b> Annotate scATAC-seq cells via label transfer</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Hands-on Analysis on Single Cell Multiome</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="data-integration" class="section level1 hasAnchor" number="8">
<h1><span class="header-section-number">Chapter 8</span> Data integration<a href="data-integration.html#data-integration" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>The PBMC multiome data was built in the <code>SeuratData</code> package and was stored in a SeuratObject already, so we can download the data and directly start analyzing. This data only needs to be loaded once, and R will skip this step the next time you load it.</p>
<pre><code>## Warning: The following packages are already installed and will not be
## reinstalled: pbmcMultiome</code></pre>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="data-integration.html#cb2-1" tabindex="-1"></a><span class="co"># load both modalities</span></span>
<span id="cb2-2"><a href="data-integration.html#cb2-2" tabindex="-1"></a>pbmc.rna <span class="ot">&lt;-</span> <span class="fu">LoadData</span>(<span class="st">&quot;pbmcMultiome&quot;</span>, <span class="st">&quot;pbmc.rna&quot;</span>)</span>
<span id="cb2-3"><a href="data-integration.html#cb2-3" tabindex="-1"></a>pbmc.atac <span class="ot">&lt;-</span> <span class="fu">LoadData</span>(<span class="st">&quot;pbmcMultiome&quot;</span>, <span class="st">&quot;pbmc.atac&quot;</span>)</span>
<span id="cb2-4"><a href="data-integration.html#cb2-4" tabindex="-1"></a></span>
<span id="cb2-5"><a href="data-integration.html#cb2-5" tabindex="-1"></a>pbmc.rna[[<span class="st">&quot;RNA&quot;</span>]] <span class="ot">&lt;-</span> <span class="fu">as</span>(pbmc.rna[[<span class="st">&quot;RNA&quot;</span>]], <span class="at">Class =</span> <span class="st">&quot;Assay5&quot;</span>)</span>
<span id="cb2-6"><a href="data-integration.html#cb2-6" tabindex="-1"></a><span class="co"># repeat QC steps performed in the WNN vignette</span></span>
<span id="cb2-7"><a href="data-integration.html#cb2-7" tabindex="-1"></a>pbmc.rna <span class="ot">&lt;-</span> <span class="fu">subset</span>(pbmc.rna, seurat_annotations <span class="sc">!=</span> <span class="st">&quot;filtered&quot;</span>)</span>
<span id="cb2-8"><a href="data-integration.html#cb2-8" tabindex="-1"></a>pbmc.atac <span class="ot">&lt;-</span> <span class="fu">subset</span>(pbmc.atac, seurat_annotations <span class="sc">!=</span> <span class="st">&quot;filtered&quot;</span>)</span>
<span id="cb2-9"><a href="data-integration.html#cb2-9" tabindex="-1"></a></span>
<span id="cb2-10"><a href="data-integration.html#cb2-10" tabindex="-1"></a><span class="co"># Perform standard analysis of each modality independently RNA analysis</span></span>
<span id="cb2-11"><a href="data-integration.html#cb2-11" tabindex="-1"></a>pbmc.rna <span class="ot">&lt;-</span> <span class="fu">NormalizeData</span>(pbmc.rna)</span>
<span id="cb2-12"><a href="data-integration.html#cb2-12" tabindex="-1"></a>pbmc.rna <span class="ot">&lt;-</span> <span class="fu">FindVariableFeatures</span>(pbmc.rna)</span>
<span id="cb2-13"><a href="data-integration.html#cb2-13" tabindex="-1"></a>pbmc.rna <span class="ot">&lt;-</span> <span class="fu">ScaleData</span>(pbmc.rna)</span>
<span id="cb2-14"><a href="data-integration.html#cb2-14" tabindex="-1"></a>pbmc.rna <span class="ot">&lt;-</span> <span class="fu">RunPCA</span>(pbmc.rna)</span>
<span id="cb2-15"><a href="data-integration.html#cb2-15" tabindex="-1"></a>pbmc.rna <span class="ot">&lt;-</span> <span class="fu">RunUMAP</span>(pbmc.rna, <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">30</span>)</span>
<span id="cb2-16"><a href="data-integration.html#cb2-16" tabindex="-1"></a></span>
<span id="cb2-17"><a href="data-integration.html#cb2-17" tabindex="-1"></a><span class="co"># ATAC analysis add gene annotation information</span></span>
<span id="cb2-18"><a href="data-integration.html#cb2-18" tabindex="-1"></a>annotations <span class="ot">&lt;-</span> <span class="fu">GetGRangesFromEnsDb</span>(<span class="at">ensdb =</span> EnsDb.Hsapiens.v86)</span>
<span id="cb2-19"><a href="data-integration.html#cb2-19" tabindex="-1"></a><span class="fu">seqlevelsStyle</span>(annotations) <span class="ot">&lt;-</span> <span class="st">&quot;UCSC&quot;</span></span>
<span id="cb2-20"><a href="data-integration.html#cb2-20" tabindex="-1"></a><span class="fu">genome</span>(annotations) <span class="ot">&lt;-</span> <span class="st">&quot;hg38&quot;</span></span>
<span id="cb2-21"><a href="data-integration.html#cb2-21" tabindex="-1"></a><span class="fu">Annotation</span>(pbmc.atac) <span class="ot">&lt;-</span> annotations</span>
<span id="cb2-22"><a href="data-integration.html#cb2-22" tabindex="-1"></a></span>
<span id="cb2-23"><a href="data-integration.html#cb2-23" tabindex="-1"></a><span class="co"># We exclude the first dimension as this is typically correlated with sequencing depth</span></span>
<span id="cb2-24"><a href="data-integration.html#cb2-24" tabindex="-1"></a>pbmc.atac <span class="ot">&lt;-</span> <span class="fu">RunTFIDF</span>(pbmc.atac)</span>
<span id="cb2-25"><a href="data-integration.html#cb2-25" tabindex="-1"></a>pbmc.atac <span class="ot">&lt;-</span> <span class="fu">FindTopFeatures</span>(pbmc.atac, <span class="at">min.cutoff =</span> <span class="st">&quot;q0&quot;</span>)</span>
<span id="cb2-26"><a href="data-integration.html#cb2-26" tabindex="-1"></a>pbmc.atac <span class="ot">&lt;-</span> <span class="fu">RunSVD</span>(pbmc.atac)</span>
<span id="cb2-27"><a href="data-integration.html#cb2-27" tabindex="-1"></a>pbmc.atac <span class="ot">&lt;-</span> <span class="fu">RunUMAP</span>(pbmc.atac, <span class="at">reduction =</span> <span class="st">&quot;lsi&quot;</span>, <span class="at">dims =</span> <span class="dv">2</span><span class="sc">:</span><span class="dv">30</span>, <span class="at">reduction.name =</span> <span class="st">&quot;umap.atac&quot;</span>, <span class="at">reduction.key =</span> <span class="st">&quot;atacUMAP_&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="data-integration.html#cb3-1" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> <span class="fu">DimPlot</span>(pbmc.rna, <span class="at">group.by =</span> <span class="st">&quot;seurat_annotations&quot;</span>, <span class="at">reduction =</span> <span class="st">&quot;umap&quot;</span>,<span class="at">label =</span> <span class="cn">TRUE</span>) <span class="sc">+</span> <span class="fu">NoLegend</span>() <span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="st">&quot;RNA&quot;</span>)</span>
<span id="cb3-2"><a href="data-integration.html#cb3-2" tabindex="-1"></a>p2 <span class="ot">&lt;-</span> <span class="fu">DimPlot</span>(pbmc.atac, <span class="at">group.by =</span> <span class="st">&quot;orig.ident&quot;</span>, <span class="at">reduction=</span><span class="st">&quot;umap.atac&quot;</span>, <span class="at">label =</span> <span class="cn">FALSE</span>) <span class="sc">+</span> <span class="fu">NoLegend</span>() <span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="st">&quot;ATAC&quot;</span>)</span>
<span id="cb3-3"><a href="data-integration.html#cb3-3" tabindex="-1"></a>p1 <span class="sc">+</span> p2</span></code></pre></div>
<p><img src="bookdown-demo_files/figure-html/unnamed-chunk-4-1.png" width="672" /></p>
<div id="identifying-anchors-between-scrna-seq-and-scatac-seq-datasets" class="section level2 hasAnchor" number="8.1">
<h2><span class="header-section-number">8.1</span> Identifying anchors between scRNA-seq and scATAC-seq datasets<a href="data-integration.html#identifying-anchors-between-scrna-seq-and-scatac-seq-datasets" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>In order to identify ‘anchors’ between scRNA-seq and scATAC-seq experiments, we first generate a rough estimate of the transcriptional activity of each gene by quantifying ATAC-seq counts in the 2 kb-upstream region and gene body, using the GeneActivity() function in the Signac package. The ensuing gene activity scores from the scATAC-seq data are then used as input for canonical correlation analysis, along with the gene expression quantifications from scRNA-seq. We perform this quantification for all genes identified as being highly variable from the scRNA-seq dataset.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="data-integration.html#cb4-1" tabindex="-1"></a><span class="co"># quantify gene activity</span></span>
<span id="cb4-2"><a href="data-integration.html#cb4-2" tabindex="-1"></a>gene.activities <span class="ot">&lt;-</span> <span class="fu">GeneActivity</span>(pbmc.atac, <span class="at">assay=</span><span class="st">&quot;ATAC&quot;</span>,<span class="at">features =</span> <span class="fu">VariableFeatures</span>(pbmc.rna))</span>
<span id="cb4-3"><a href="data-integration.html#cb4-3" tabindex="-1"></a></span>
<span id="cb4-4"><a href="data-integration.html#cb4-4" tabindex="-1"></a><span class="co"># add gene activities as a new assay</span></span>
<span id="cb4-5"><a href="data-integration.html#cb4-5" tabindex="-1"></a>pbmc.atac[[<span class="st">&quot;ACTIVITY&quot;</span>]] <span class="ot">&lt;-</span> <span class="fu">CreateAssayObject</span>(<span class="at">counts =</span> gene.activities)</span>
<span id="cb4-6"><a href="data-integration.html#cb4-6" tabindex="-1"></a></span>
<span id="cb4-7"><a href="data-integration.html#cb4-7" tabindex="-1"></a><span class="co"># normalize gene activities</span></span>
<span id="cb4-8"><a href="data-integration.html#cb4-8" tabindex="-1"></a><span class="fu">DefaultAssay</span>(pbmc.atac) <span class="ot">&lt;-</span> <span class="st">&quot;ACTIVITY&quot;</span></span>
<span id="cb4-9"><a href="data-integration.html#cb4-9" tabindex="-1"></a>pbmc.atac <span class="ot">&lt;-</span> <span class="fu">NormalizeData</span>(pbmc.atac)</span>
<span id="cb4-10"><a href="data-integration.html#cb4-10" tabindex="-1"></a>pbmc.atac <span class="ot">&lt;-</span> <span class="fu">ScaleData</span>(pbmc.atac, <span class="at">features =</span> <span class="fu">rownames</span>(pbmc.atac))</span>
<span id="cb4-11"><a href="data-integration.html#cb4-11" tabindex="-1"></a></span>
<span id="cb4-12"><a href="data-integration.html#cb4-12" tabindex="-1"></a><span class="co"># Identify anchors</span></span>
<span id="cb4-13"><a href="data-integration.html#cb4-13" tabindex="-1"></a>transfer.anchors <span class="ot">&lt;-</span> <span class="fu">FindTransferAnchors</span>(<span class="at">reference =</span> pbmc.rna, <span class="at">query =</span> pbmc.atac, <span class="at">features =</span> <span class="fu">VariableFeatures</span>(<span class="at">object =</span> pbmc.rna),</span>
<span id="cb4-14"><a href="data-integration.html#cb4-14" tabindex="-1"></a>    <span class="at">reference.assay =</span> <span class="st">&quot;RNA&quot;</span>, <span class="at">query.assay =</span> <span class="st">&quot;ACTIVITY&quot;</span>, <span class="at">reduction =</span> <span class="st">&quot;cca&quot;</span>)</span></code></pre></div>
</div>
<div id="annotate-scatac-seq-cells-via-label-transfer" class="section level2 hasAnchor" number="8.2">
<h2><span class="header-section-number">8.2</span> Annotate scATAC-seq cells via label transfer<a href="data-integration.html#annotate-scatac-seq-cells-via-label-transfer" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>After identifying anchors, we can transfer annotations from the scRNA-seq dataset onto the scATAC-seq cells. The annotations are stored in the seurat_annotations field, and are provided as input to the refdata parameter. The output will contain a matrix with predictions and confidence scores for each ATAC-seq cell.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="data-integration.html#cb5-1" tabindex="-1"></a>celltype.predictions <span class="ot">&lt;-</span> <span class="fu">TransferData</span>(<span class="at">anchorset =</span> transfer.anchors, <span class="at">refdata =</span> pbmc.rna<span class="sc">$</span>seurat_annotations,</span>
<span id="cb5-2"><a href="data-integration.html#cb5-2" tabindex="-1"></a>    <span class="at">weight.reduction =</span> pbmc.atac[[<span class="st">&quot;lsi&quot;</span>]], <span class="at">dims =</span> <span class="dv">2</span><span class="sc">:</span><span class="dv">30</span>)</span>
<span id="cb5-3"><a href="data-integration.html#cb5-3" tabindex="-1"></a></span>
<span id="cb5-4"><a href="data-integration.html#cb5-4" tabindex="-1"></a>pbmc.atac <span class="ot">&lt;-</span> <span class="fu">AddMetaData</span>(pbmc.atac, <span class="at">metadata =</span> celltype.predictions)</span></code></pre></div>
<p>After performing transfer, the ATAC-seq cells have predicted annotations (transferred from the scRNA-seq dataset) stored in the predicted.id field. Since these cells were measured with the multiome kit, we also have a ground-truth annotation that can be used for evaluation. You can see that the predicted and actual annotations are extremely similar.</p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-8"></span>
<img src="bookdown-demo_files/figure-html/unnamed-chunk-8-1.png" alt="Compare cell labels in integrated vs truth" width="100%" />
<p class="caption">
Figure 8.1: Compare cell labels in integrated vs truth
</p>
</div>
<p>In this example, the annotation for an scATAC-seq profile is correctly predicted via scRNA-seq integration ~90% of the time. In addition, the prediction.score.max field quantifies the uncertainty associated with our predicted annotations. We can see that cells that are correctly annotated are typically associated with high prediction scores (&gt;90%), while cells that are incorrectly annotated are associated with sharply lower prediction scores (&lt;50%). Incorrect assignments also tend to reflect closely related cell types (i.e. Intermediate vs. Naive B cells).</p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-9"></span>
<img src="bookdown-demo_files/figure-html/unnamed-chunk-9-1.png" alt="Prediction accuracy" width="150%" />
<p class="caption">
Figure 8.2: Prediction accuracy
</p>
</div>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="data-integration.html#cb6-1" tabindex="-1"></a><span class="fu">saveRDS</span>(pbmc.atac,<span class="at">file=</span><span class="st">&quot;pbmc_atac_integrated.rds&quot;</span>)</span></code></pre></div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="peak-calling-and-motif-analysis.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="references.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/07-integration.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["bookdown-demo.pdf", "bookdown-demo.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
