<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 6 Differential Expression (DE) analysis and cell type annotation | Hands-on Analysis on Single Cell Multiome</title>
  <meta name="description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  <meta name="generator" content="bookdown 0.39 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 6 Differential Expression (DE) analysis and cell type annotation | Hands-on Analysis on Single Cell Multiome" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 6 Differential Expression (DE) analysis and cell type annotation | Hands-on Analysis on Single Cell Multiome" />
  
  <meta name="twitter:description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  

<meta name="author" content="Danni Liu" />


<meta name="date" content="2024-06-07" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="dimension-reduction-and-clustering.html"/>
<link rel="next" href="peak-calling-and-motif-analysis.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Hands-on Analysis on Single Cell Multiome</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Prerequisites</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#about-data"><i class="fa fa-check"></i><b>1.1</b> About Data</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#server-environment-requirements"><i class="fa fa-check"></i><b>1.2</b> Server environment requirements</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#r-libraries"><i class="fa fa-check"></i><b>1.3</b> R libraries</a></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#construct-folder-structure-in-terminal"><i class="fa fa-check"></i><b>1.4</b> Construct folder structure in Terminal</a></li>
<li class="chapter" data-level="1.5" data-path="index.html"><a href="index.html#accessing-pancreas-data"><i class="fa fa-check"></i><b>1.5</b> Accessing pancreas data</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="cellranger_arc.html"><a href="cellranger_arc.html"><i class="fa fa-check"></i><b>2</b> Cell Ranger ARC for preprocessing sequencing data</a>
<ul>
<li class="chapter" data-level="2.1" data-path="cellranger_arc.html"><a href="cellranger_arc.html#cellranger-arc-mkfastq"><i class="fa fa-check"></i><b>2.1</b> <code>cellranger-arc mkfastq</code></a></li>
<li class="chapter" data-level="2.2" data-path="cellranger_arc.html"><a href="cellranger_arc.html#bcl-naming-convention"><i class="fa fa-check"></i><b>2.2</b> BCL naming convention</a></li>
<li class="chapter" data-level="2.3" data-path="cellranger_arc.html"><a href="cellranger_arc.html#cellranger-arc-count"><i class="fa fa-check"></i><b>2.3</b> <code>cellranger-arc count</code></a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="seuratqc.html"><a href="seuratqc.html"><i class="fa fa-check"></i><b>3</b> Obtain data and perform QC using Seurat</a>
<ul>
<li class="chapter" data-level="3.1" data-path="seuratqc.html"><a href="seuratqc.html#download-data"><i class="fa fa-check"></i><b>3.1</b> Download data</a></li>
<li class="chapter" data-level="3.2" data-path="seuratqc.html"><a href="seuratqc.html#load-data-in-r-and-construct-seurat-object"><i class="fa fa-check"></i><b>3.2</b> Load data in R and construct Seurat object</a></li>
<li class="chapter" data-level="3.3" data-path="seuratqc.html"><a href="seuratqc.html#perform-quality-control-using-both-assays"><i class="fa fa-check"></i><b>3.3</b> Perform quality control using both assays</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="normalization-and-cell-cycle-correction.html"><a href="normalization-and-cell-cycle-correction.html"><i class="fa fa-check"></i><b>4</b> Normalization and cell cycle correction</a>
<ul>
<li class="chapter" data-level="4.1" data-path="normalization-and-cell-cycle-correction.html"><a href="normalization-and-cell-cycle-correction.html#normalization"><i class="fa fa-check"></i><b>4.1</b> Normalization</a></li>
<li class="chapter" data-level="4.2" data-path="normalization-and-cell-cycle-correction.html"><a href="normalization-and-cell-cycle-correction.html#cell-cycle-correction-and-scale-data"><i class="fa fa-check"></i><b>4.2</b> Cell cycle correction and scale data</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="dimension-reduction-and-clustering.html"><a href="dimension-reduction-and-clustering.html"><i class="fa fa-check"></i><b>5</b> Dimension reduction and clustering</a>
<ul>
<li class="chapter" data-level="5.1" data-path="dimension-reduction-and-clustering.html"><a href="dimension-reduction-and-clustering.html#identify-top-variable-features"><i class="fa fa-check"></i><b>5.1</b> Identify top variable features</a></li>
<li class="chapter" data-level="5.2" data-path="dimension-reduction-and-clustering.html"><a href="dimension-reduction-and-clustering.html#dimension-reduction"><i class="fa fa-check"></i><b>5.2</b> Dimension reduction</a></li>
<li class="chapter" data-level="5.3" data-path="dimension-reduction-and-clustering.html"><a href="dimension-reduction-and-clustering.html#clustering-and-joint-clustering"><i class="fa fa-check"></i><b>5.3</b> Clustering and joint clustering</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="differential-expression-de-analysis-and-cell-type-annotation.html"><a href="differential-expression-de-analysis-and-cell-type-annotation.html"><i class="fa fa-check"></i><b>6</b> Differential Expression (DE) analysis and cell type annotation</a>
<ul>
<li class="chapter" data-level="6.1" data-path="differential-expression-de-analysis-and-cell-type-annotation.html"><a href="differential-expression-de-analysis-and-cell-type-annotation.html#de-analysis"><i class="fa fa-check"></i><b>6.1</b> DE analysis</a></li>
<li class="chapter" data-level="6.2" data-path="differential-expression-de-analysis-and-cell-type-annotation.html"><a href="differential-expression-de-analysis-and-cell-type-annotation.html#cell-type-annotation"><i class="fa fa-check"></i><b>6.2</b> Cell type annotation</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="peak-calling-and-motif-analysis.html"><a href="peak-calling-and-motif-analysis.html"><i class="fa fa-check"></i><b>7</b> Peak calling and Motif analysis</a>
<ul>
<li class="chapter" data-level="7.1" data-path="peak-calling-and-motif-analysis.html"><a href="peak-calling-and-motif-analysis.html#linkage-plot"><i class="fa fa-check"></i><b>7.1</b> Linkage plot</a></li>
<li class="chapter" data-level="7.2" data-path="peak-calling-and-motif-analysis.html"><a href="peak-calling-and-motif-analysis.html#call-peaks-using-macs2"><i class="fa fa-check"></i><b>7.2</b> Call peaks using MACS2</a></li>
<li class="chapter" data-level="7.3" data-path="peak-calling-and-motif-analysis.html"><a href="peak-calling-and-motif-analysis.html#motif-analysis"><i class="fa fa-check"></i><b>7.3</b> Motif analysis</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="data-integration.html"><a href="data-integration.html"><i class="fa fa-check"></i><b>8</b> Data integration</a>
<ul>
<li class="chapter" data-level="8.1" data-path="data-integration.html"><a href="data-integration.html#identifying-anchors-between-scrna-seq-and-scatac-seq-datasets"><i class="fa fa-check"></i><b>8.1</b> Identifying anchors between scRNA-seq and scATAC-seq datasets</a></li>
<li class="chapter" data-level="8.2" data-path="data-integration.html"><a href="data-integration.html#annotate-scatac-seq-cells-via-label-transfer"><i class="fa fa-check"></i><b>8.2</b> Annotate scATAC-seq cells via label transfer</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Hands-on Analysis on Single Cell Multiome</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="differential-expression-de-analysis-and-cell-type-annotation" class="section level1 hasAnchor" number="6">
<h1><span class="header-section-number">Chapter 6</span> Differential Expression (DE) analysis and cell type annotation<a href="differential-expression-de-analysis-and-cell-type-annotation.html#differential-expression-de-analysis-and-cell-type-annotation" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>Continue from checkpoint…</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="differential-expression-de-analysis-and-cell-type-annotation.html#cb1-1" tabindex="-1"></a>pbmc<span class="ot">&lt;-</span><span class="fu">readRDS</span>(<span class="st">&quot;../pbmc_data/pbmc_clust.rds&quot;</span>)</span></code></pre></div>
<div id="de-analysis" class="section level2 hasAnchor" number="6.1">
<h2><span class="header-section-number">6.1</span> DE analysis<a href="differential-expression-de-analysis-and-cell-type-annotation.html#de-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>DE analysis identify gene markers that are differentially expressed in one cell type vs all others. You can perform DE for one cell type using <code>FindMarkers</code>, or run for all clusters using <code>FindAllMarkers</code>.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="differential-expression-de-analysis-and-cell-type-annotation.html#cb2-1" tabindex="-1"></a><span class="fu">DefaultAssay</span>(pbmc) <span class="ot">&lt;-</span> <span class="st">&quot;RNA&quot;</span></span>
<span id="cb2-2"><a href="differential-expression-de-analysis-and-cell-type-annotation.html#cb2-2" tabindex="-1"></a><span class="co"># cluster 0 vs all others</span></span>
<span id="cb2-3"><a href="differential-expression-de-analysis-and-cell-type-annotation.html#cb2-3" tabindex="-1"></a><span class="co"># this can take several minutes</span></span>
<span id="cb2-4"><a href="differential-expression-de-analysis-and-cell-type-annotation.html#cb2-4" tabindex="-1"></a>cluster0.markers <span class="ot">&lt;-</span> <span class="fu">FindMarkers</span>(<span class="at">object =</span> pbmc, <span class="at">ident.1 =</span> <span class="dv">0</span>, <span class="at">min.pct =</span> <span class="fl">0.1</span>)</span>
<span id="cb2-5"><a href="differential-expression-de-analysis-and-cell-type-annotation.html#cb2-5" tabindex="-1"></a><span class="fu">print</span>(<span class="at">x =</span> <span class="fu">head</span>(<span class="at">x =</span> cluster0.markers, <span class="at">n =</span> <span class="dv">5</span>))</span></code></pre></div>
<pre><code>##        p_val avg_log2FC pct.1 pct.2 p_val_adj
## LRMDA      0   3.118612 0.971 0.177         0
## PLXDC2     0   3.284448 0.993 0.204         0
## SLC8A1     0   2.868579 0.974 0.193         0
## CSF3R      0   3.534311 0.893 0.128         0
## DMXL2      0   3.023065 0.921 0.179         0</code></pre>
<p>In the code below, we can also use <code>FindAllMarkers</code> function to automate this process for all clusters. This compares 0 with all other clusters, 1 with all other clusters, etc. Note here that we have specified that min.pct =0.25 , as we want the log(fold-change) threshold for a gene to be at least 0.25 to be identified as significant. Then we select the top 3 markers for each cluster, based on log(fold-changes, but it will take long time to finish.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="differential-expression-de-analysis-and-cell-type-annotation.html#cb4-1" tabindex="-1"></a><span class="fu">DefaultAssay</span>(pbmc) <span class="ot">&lt;-</span> <span class="st">&quot;RNA&quot;</span></span>
<span id="cb4-2"><a href="differential-expression-de-analysis-and-cell-type-annotation.html#cb4-2" tabindex="-1"></a><span class="co"># find markers for every cluster compared to all remaining cells, report only the positive ones</span></span>
<span id="cb4-3"><a href="differential-expression-de-analysis-and-cell-type-annotation.html#cb4-3" tabindex="-1"></a><span class="co"># FindAllMarkers can take long time to run</span></span>
<span id="cb4-4"><a href="differential-expression-de-analysis-and-cell-type-annotation.html#cb4-4" tabindex="-1"></a>allmarkers <span class="ot">&lt;-</span><span class="fu">FindAllMarkers</span>(pbmc, <span class="at">only.pos =</span> <span class="cn">FALSE</span>, <span class="at">min.pct =</span> <span class="fl">0.25</span>, <span class="at">logfc.threshold =</span> <span class="fl">0.25</span>,<span class="at">assay=</span><span class="st">&quot;RNA&quot;</span>,<span class="at">slot=</span><span class="st">&quot;scale.data&quot;</span>)</span>
<span id="cb4-5"><a href="differential-expression-de-analysis-and-cell-type-annotation.html#cb4-5" tabindex="-1"></a></span>
<span id="cb4-6"><a href="differential-expression-de-analysis-and-cell-type-annotation.html#cb4-6" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb4-7"><a href="differential-expression-de-analysis-and-cell-type-annotation.html#cb4-7" tabindex="-1"></a>allmarkers_grouped<span class="ot">&lt;-</span> allmarkers <span class="sc">%&gt;%</span></span>
<span id="cb4-8"><a href="differential-expression-de-analysis-and-cell-type-annotation.html#cb4-8" tabindex="-1"></a>    <span class="fu">group_by</span>(cluster) <span class="sc">%&gt;%</span> <span class="fu">slice_max</span>(<span class="at">n =</span> <span class="dv">1</span>, <span class="at">order_by =</span> avg_log2FC)</span>
<span id="cb4-9"><a href="differential-expression-de-analysis-and-cell-type-annotation.html#cb4-9" tabindex="-1"></a><span class="fu">write.table</span>(<span class="fu">as.data.frame</span>(allmarkers),<span class="at">file=</span><span class="st">&quot;../pbmc_data/DE_markers.txt&quot;</span>,<span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>,<span class="at">row.names =</span> F,<span class="at">col.names =</span> T)</span></code></pre></div>
<p>Visualizing DE markers.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="differential-expression-de-analysis-and-cell-type-annotation.html#cb5-1" tabindex="-1"></a>allmarkers<span class="ot">&lt;-</span><span class="fu">read.table</span>(<span class="st">&quot;../pbmc_data/DE_markers.txt&quot;</span>,<span class="at">header=</span>T)</span>
<span id="cb5-2"><a href="differential-expression-de-analysis-and-cell-type-annotation.html#cb5-2" tabindex="-1"></a>top3_log2FC <span class="ot">&lt;-</span> allmarkers <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(cluster) <span class="sc">%&gt;%</span> <span class="fu">top_n</span>(<span class="dv">3</span>, avg_log2FC)</span>
<span id="cb5-3"><a href="differential-expression-de-analysis-and-cell-type-annotation.html#cb5-3" tabindex="-1"></a><span class="fu">head</span>(top3_log2FC)</span></code></pre></div>
<pre><code>## # A tibble: 6 × 7
## # Groups:   cluster [2]
##       p_val avg_log2FC pct.1 pct.2 p_val_adj cluster gene      
##       &lt;dbl&gt;      &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;     &lt;dbl&gt; &lt;chr&gt;   &lt;chr&gt;     
## 1 6.83e- 15       3.46 0.275 0.04  2.50e- 10 1       ANKRD55   
## 2 1.01e- 14       2.59 0.499 0.126 3.69e- 10 1       AC139720.1
## 3 1.76e-  6       2.91 0.444 0.096 6.46e-  2 1       TSHZ2     
## 4 4.20e-156       2.22 0.895 0.372 1.54e-151 5       ITGB1     
## 5 6.84e- 13       2.26 0.484 0.123 2.50e-  8 5       GATA3     
## 6 1.47e- 12       2.15 0.461 0.133 5.40e-  8 5       MFHAS1</code></pre>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="differential-expression-de-analysis-and-cell-type-annotation.html#cb7-1" tabindex="-1"></a><span class="co"># plot</span></span>
<span id="cb7-2"><a href="differential-expression-de-analysis-and-cell-type-annotation.html#cb7-2" tabindex="-1"></a>features<span class="ot">&lt;-</span><span class="fu">unique</span>(top3_log2FC<span class="sc">$</span>gene)</span>
<span id="cb7-3"><a href="differential-expression-de-analysis-and-cell-type-annotation.html#cb7-3" tabindex="-1"></a><span class="fu">VlnPlot</span>(pbmc, <span class="at">features =</span> <span class="fu">c</span>(<span class="st">&quot;TSHZ2&quot;</span>,<span class="st">&quot;LEF1&quot;</span>),<span class="at">slot =</span> <span class="st">&quot;counts&quot;</span>, <span class="at">log =</span> <span class="cn">TRUE</span>, <span class="at">pt.size=</span><span class="dv">0</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-5"></span>
<img src="bookdown-demo_files/figure-html/unnamed-chunk-5-1.png" alt="Violin plot for gene TSHZ2 and LEF1" width="100%" />
<p class="caption">
Figure 6.1: Violin plot for gene TSHZ2 and LEF1
</p>
</div>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="differential-expression-de-analysis-and-cell-type-annotation.html#cb8-1" tabindex="-1"></a><span class="fu">FeaturePlot</span>(pbmc, <span class="at">features =</span> <span class="fu">c</span>(<span class="st">&quot;LEF1&quot;</span>, <span class="st">&quot;FHIT&quot;</span>, <span class="st">&quot;LTB&quot;</span>, <span class="st">&quot;NELL2&quot;</span>, <span class="st">&quot;AFF3&quot;</span>, <span class="st">&quot;BANK1&quot;</span>, <span class="st">&quot;VCAN&quot;</span>, <span class="st">&quot;FCGR3A&quot;</span>, <span class="st">&quot;TCF4&quot;</span>))</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-6"></span>
<img src="bookdown-demo_files/figure-html/unnamed-chunk-6-1.png" alt="Feature expression plot" width="100%" />
<p class="caption">
Figure 6.2: Feature expression plot
</p>
</div>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="differential-expression-de-analysis-and-cell-type-annotation.html#cb9-1" tabindex="-1"></a><span class="fu">DoHeatmap</span>(<span class="at">object =</span> pbmc, <span class="at">features =</span> top3_log2FC<span class="sc">$</span>gene)<span class="sc">+</span> <span class="fu">NoLegend</span>()</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-7"></span>
<img src="bookdown-demo_files/figure-html/unnamed-chunk-7-1.png" alt="Heatmap of top 3 DE markers for each cluster" width="400%" />
<p class="caption">
Figure 6.3: Heatmap of top 3 DE markers for each cluster
</p>
</div>
</div>
<div id="cell-type-annotation" class="section level2 hasAnchor" number="6.2">
<h2><span class="header-section-number">6.2</span> Cell type annotation<a href="differential-expression-de-analysis-and-cell-type-annotation.html#cell-type-annotation" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We annotate the clusters below. Note that you could also annotate the dataset using supervised mapping pipelines or automated web tool, <a href="https://satijalab.org/azimuth/">Azimuth</a>.</p>
<p>To annotate cell types in the dataset we can transfer cell labels from an existing PBMC reference dataset using tools in the Seurat package. See the Seurat reference mapping vignette for more information.</p>
<p>We’ll use an annotated PBMC reference dataset from Hao et al. (2020), available for download here: <a href="https://atlas.fredhutch.org/data/nygc/multimodal/pbmc_multimodal.h5seurat" class="uri">https://atlas.fredhutch.org/data/nygc/multimodal/pbmc_multimodal.h5seurat</a></p>
<p>Note that the SeuratDisk package is required to load the reference dataset.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb10-1"><a href="differential-expression-de-analysis-and-cell-type-annotation.html#cb10-1" tabindex="-1"></a><span class="co"># download reference annotation</span></span>
<span id="cb10-2"><a href="differential-expression-de-analysis-and-cell-type-annotation.html#cb10-2" tabindex="-1"></a><span class="fu">wget</span> https://atlas.fredhutch.org/data/nygc/multimodal/pbmc_multimodal.h5seurat</span></code></pre></div>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="differential-expression-de-analysis-and-cell-type-annotation.html#cb11-1" tabindex="-1"></a><span class="fu">library</span>(SeuratDisk)</span>
<span id="cb11-2"><a href="differential-expression-de-analysis-and-cell-type-annotation.html#cb11-2" tabindex="-1"></a><span class="co"># load PBMC reference</span></span>
<span id="cb11-3"><a href="differential-expression-de-analysis-and-cell-type-annotation.html#cb11-3" tabindex="-1"></a>reference <span class="ot">&lt;-</span> <span class="fu">LoadH5Seurat</span>(<span class="st">&quot;../pbmc_data/pbmc_multimodal.h5seurat&quot;</span>)</span>
<span id="cb11-4"><a href="differential-expression-de-analysis-and-cell-type-annotation.html#cb11-4" tabindex="-1"></a><span class="fu">DefaultAssay</span>(pbmc) <span class="ot">&lt;-</span> <span class="st">&quot;SCT&quot;</span></span>
<span id="cb11-5"><a href="differential-expression-de-analysis-and-cell-type-annotation.html#cb11-5" tabindex="-1"></a></span>
<span id="cb11-6"><a href="differential-expression-de-analysis-and-cell-type-annotation.html#cb11-6" tabindex="-1"></a><span class="co"># transfer cell type labels from reference to query</span></span>
<span id="cb11-7"><a href="differential-expression-de-analysis-and-cell-type-annotation.html#cb11-7" tabindex="-1"></a>transfer_anchors <span class="ot">&lt;-</span> <span class="fu">FindTransferAnchors</span>(</span>
<span id="cb11-8"><a href="differential-expression-de-analysis-and-cell-type-annotation.html#cb11-8" tabindex="-1"></a>  <span class="at">reference =</span> reference, <span class="at">query =</span> pbmc, </span>
<span id="cb11-9"><a href="differential-expression-de-analysis-and-cell-type-annotation.html#cb11-9" tabindex="-1"></a>    <span class="at">reference.reduction=</span><span class="st">&quot;spca&quot;</span>, <span class="at">normalization.method =</span> <span class="st">&quot;SCT&quot;</span>, </span>
<span id="cb11-10"><a href="differential-expression-de-analysis-and-cell-type-annotation.html#cb11-10" tabindex="-1"></a>    <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">50</span> )</span></code></pre></div>
<p>MapQuery() is a wrapper around three functions: TransferData(), IntegrateEmbeddings(), and ProjectUMAP(). TransferData() is used to transfer cell type labels and impute the ADT values. IntegrateEmbeddings() and ProjectUMAP() are used to project the query data onto the UMAP structure of the reference.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="differential-expression-de-analysis-and-cell-type-annotation.html#cb12-1" tabindex="-1"></a><span class="co"># MapQuery() is a wrapper around three functions: TransferData(), IntegrateEmbeddings(), and ProjectUMAP()</span></span>
<span id="cb12-2"><a href="differential-expression-de-analysis-and-cell-type-annotation.html#cb12-2" tabindex="-1"></a>pbmc <span class="ot">&lt;-</span> <span class="fu">MapQuery</span>(</span>
<span id="cb12-3"><a href="differential-expression-de-analysis-and-cell-type-annotation.html#cb12-3" tabindex="-1"></a>  <span class="at">anchorset =</span> transfer_anchors,</span>
<span id="cb12-4"><a href="differential-expression-de-analysis-and-cell-type-annotation.html#cb12-4" tabindex="-1"></a>  <span class="at">query =</span> pbmc,</span>
<span id="cb12-5"><a href="differential-expression-de-analysis-and-cell-type-annotation.html#cb12-5" tabindex="-1"></a>  <span class="at">reference =</span> reference,</span>
<span id="cb12-6"><a href="differential-expression-de-analysis-and-cell-type-annotation.html#cb12-6" tabindex="-1"></a>  <span class="at">refdata =</span> <span class="fu">list</span>(</span>
<span id="cb12-7"><a href="differential-expression-de-analysis-and-cell-type-annotation.html#cb12-7" tabindex="-1"></a>    <span class="at">celltype =</span> <span class="st">&quot;celltype.l2&quot;</span>,</span>
<span id="cb12-8"><a href="differential-expression-de-analysis-and-cell-type-annotation.html#cb12-8" tabindex="-1"></a>    <span class="at">predicted_ADT =</span> <span class="st">&quot;ADT&quot;</span></span>
<span id="cb12-9"><a href="differential-expression-de-analysis-and-cell-type-annotation.html#cb12-9" tabindex="-1"></a>  ),</span>
<span id="cb12-10"><a href="differential-expression-de-analysis-and-cell-type-annotation.html#cb12-10" tabindex="-1"></a>  <span class="at">reference.reduction =</span> <span class="st">&quot;spca&quot;</span>, <span class="co"># reference has different types of reduction performed, pca, spca, umap ,etc.</span></span>
<span id="cb12-11"><a href="differential-expression-de-analysis-and-cell-type-annotation.html#cb12-11" tabindex="-1"></a>  <span class="at">reduction.model =</span> <span class="st">&quot;wnn.umap&quot;</span>,</span>
<span id="cb12-12"><a href="differential-expression-de-analysis-and-cell-type-annotation.html#cb12-12" tabindex="-1"></a>    <span class="at">transferdata.args=</span><span class="fu">list</span>(),</span>
<span id="cb12-13"><a href="differential-expression-de-analysis-and-cell-type-annotation.html#cb12-13" tabindex="-1"></a>    <span class="at">integrateembeddings.args =</span> <span class="fu">list</span>(),</span>
<span id="cb12-14"><a href="differential-expression-de-analysis-and-cell-type-annotation.html#cb12-14" tabindex="-1"></a>  <span class="at">projectumap.args =</span> <span class="fu">list</span>()</span>
<span id="cb12-15"><a href="differential-expression-de-analysis-and-cell-type-annotation.html#cb12-15" tabindex="-1"></a>)</span></code></pre></div>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="differential-expression-de-analysis-and-cell-type-annotation.html#cb13-1" tabindex="-1"></a><span class="co"># plot on reference data space</span></span>
<span id="cb13-2"><a href="differential-expression-de-analysis-and-cell-type-annotation.html#cb13-2" tabindex="-1"></a>p1<span class="ot">&lt;-</span><span class="fu">DimPlot</span>(pbmc, <span class="at">reduction =</span> <span class="st">&quot;ref.umap&quot;</span>, <span class="at">group.by =</span> <span class="st">&quot;predicted.celltype.l2&quot;</span>, <span class="at">label =</span> <span class="cn">TRUE</span>, <span class="at">label.size =</span> <span class="dv">3</span> ,<span class="at">repel =</span> <span class="cn">TRUE</span>)</span>
<span id="cb13-3"><a href="differential-expression-de-analysis-and-cell-type-annotation.html#cb13-3" tabindex="-1"></a><span class="co"># plot on wnn space</span></span>
<span id="cb13-4"><a href="differential-expression-de-analysis-and-cell-type-annotation.html#cb13-4" tabindex="-1"></a>p2<span class="ot">&lt;-</span><span class="fu">DimPlot</span>(pbmc, <span class="at">reduction =</span> <span class="st">&quot;wnn.umap&quot;</span>, <span class="at">group.by =</span> <span class="st">&quot;predicted.celltype.l2&quot;</span>, <span class="at">label =</span> <span class="cn">TRUE</span>, <span class="at">label.size =</span> <span class="dv">3</span> ,<span class="at">repel =</span> <span class="cn">TRUE</span>)</span>
<span id="cb13-5"><a href="differential-expression-de-analysis-and-cell-type-annotation.html#cb13-5" tabindex="-1"></a>p1 <span class="sc">+</span> p2 <span class="sc">&amp;</span> <span class="fu">NoLegend</span>() <span class="sc">&amp;</span> <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="fl">0.5</span>))</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-12"></span>
<img src="bookdown-demo_files/figure-html/unnamed-chunk-12-1.png" alt="Clusters for annotated cell types" width="150%" />
<p class="caption">
Figure 6.4: Clusters for annotated cell types
</p>
</div>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="differential-expression-de-analysis-and-cell-type-annotation.html#cb14-1" tabindex="-1"></a><span class="fu">saveRDS</span>(pbmc,<span class="at">file=</span><span class="st">&quot;pbmc_anno.rds&quot;</span>)</span></code></pre></div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="dimension-reduction-and-clustering.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="peak-calling-and-motif-analysis.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/05-DEanalysis.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["bookdown-demo.pdf", "bookdown-demo.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
