<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 5 Dimension reduction and clustering | Hands-on Analysis on Single Cell Multiome</title>
  <meta name="description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  <meta name="generator" content="bookdown 0.39 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 5 Dimension reduction and clustering | Hands-on Analysis on Single Cell Multiome" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 5 Dimension reduction and clustering | Hands-on Analysis on Single Cell Multiome" />
  
  <meta name="twitter:description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  

<meta name="author" content="Danni Liu" />


<meta name="date" content="2024-06-05" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="normalization-and-cell-cycle-correction.html"/>
<link rel="next" href="differential-expression-de-analysis-and-cell-type-annotation.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Hands-on Analysis on Single Cell Multiome</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Prerequisites</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#about-data"><i class="fa fa-check"></i><b>1.1</b> About Data</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#server-environment-requirements"><i class="fa fa-check"></i><b>1.2</b> Server environment requirements</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#r-libraries"><i class="fa fa-check"></i><b>1.3</b> R libraries</a></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#construct-folder-structure"><i class="fa fa-check"></i><b>1.4</b> Construct folder structure</a></li>
<li class="chapter" data-level="1.5" data-path="index.html"><a href="index.html#downloading-fastq-files"><i class="fa fa-check"></i><b>1.5</b> Downloading FASTQ files</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="cellranger_arc.html"><a href="cellranger_arc.html"><i class="fa fa-check"></i><b>2</b> Cell Ranger ARC for preprocessing sequencing data</a>
<ul>
<li class="chapter" data-level="2.1" data-path="cellranger_arc.html"><a href="cellranger_arc.html#cellranger-arc-mkfastq"><i class="fa fa-check"></i><b>2.1</b> <code>cellranger-arc mkfastq</code></a></li>
<li class="chapter" data-level="2.2" data-path="cellranger_arc.html"><a href="cellranger_arc.html#bcl-naming-convention"><i class="fa fa-check"></i><b>2.2</b> BCL naming convention</a></li>
<li class="chapter" data-level="2.3" data-path="cellranger_arc.html"><a href="cellranger_arc.html#cellranger-arc-count"><i class="fa fa-check"></i><b>2.3</b> <code>cellranger-arc count</code></a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="seuratqc.html"><a href="seuratqc.html"><i class="fa fa-check"></i><b>3</b> Obtain data and perform QC using Seurat</a>
<ul>
<li class="chapter" data-level="3.1" data-path="seuratqc.html"><a href="seuratqc.html#download-data"><i class="fa fa-check"></i><b>3.1</b> Download data</a></li>
<li class="chapter" data-level="3.2" data-path="seuratqc.html"><a href="seuratqc.html#load-data-in-r-and-construct-seurat-object"><i class="fa fa-check"></i><b>3.2</b> Load data in R and construct Seurat object</a></li>
<li class="chapter" data-level="3.3" data-path="seuratqc.html"><a href="seuratqc.html#perform-quality-control-using-both-assays"><i class="fa fa-check"></i><b>3.3</b> Perform quality control using both assays</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="normalization-and-cell-cycle-correction.html"><a href="normalization-and-cell-cycle-correction.html"><i class="fa fa-check"></i><b>4</b> Normalization and cell cycle correction</a>
<ul>
<li class="chapter" data-level="4.1" data-path="normalization-and-cell-cycle-correction.html"><a href="normalization-and-cell-cycle-correction.html#normalization"><i class="fa fa-check"></i><b>4.1</b> Normalization</a></li>
<li class="chapter" data-level="4.2" data-path="normalization-and-cell-cycle-correction.html"><a href="normalization-and-cell-cycle-correction.html#cell-cycle-correction-and-scale-data"><i class="fa fa-check"></i><b>4.2</b> Cell cycle correction and scale data</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="dimension-reduction-and-clustering.html"><a href="dimension-reduction-and-clustering.html"><i class="fa fa-check"></i><b>5</b> Dimension reduction and clustering</a>
<ul>
<li class="chapter" data-level="5.1" data-path="dimension-reduction-and-clustering.html"><a href="dimension-reduction-and-clustering.html#identify-top-variable-features"><i class="fa fa-check"></i><b>5.1</b> Identify top variable features</a></li>
<li class="chapter" data-level="5.2" data-path="dimension-reduction-and-clustering.html"><a href="dimension-reduction-and-clustering.html#dimension-reduction"><i class="fa fa-check"></i><b>5.2</b> Dimension reduction</a></li>
<li class="chapter" data-level="5.3" data-path="dimension-reduction-and-clustering.html"><a href="dimension-reduction-and-clustering.html#clustering-and-joint-clustering"><i class="fa fa-check"></i><b>5.3</b> Clustering and joint clustering</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="differential-expression-de-analysis-and-cell-type-annotation.html"><a href="differential-expression-de-analysis-and-cell-type-annotation.html"><i class="fa fa-check"></i><b>6</b> Differential Expression (DE) analysis and cell type annotation</a>
<ul>
<li class="chapter" data-level="6.1" data-path="differential-expression-de-analysis-and-cell-type-annotation.html"><a href="differential-expression-de-analysis-and-cell-type-annotation.html#de-analysis"><i class="fa fa-check"></i><b>6.1</b> DE analysis</a></li>
<li class="chapter" data-level="6.2" data-path="differential-expression-de-analysis-and-cell-type-annotation.html"><a href="differential-expression-de-analysis-and-cell-type-annotation.html#cell-type-annotation"><i class="fa fa-check"></i><b>6.2</b> Cell type annotation</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="peak-calling-and-motif-analysis.html"><a href="peak-calling-and-motif-analysis.html"><i class="fa fa-check"></i><b>7</b> Peak calling and Motif analysis</a>
<ul>
<li class="chapter" data-level="7.1" data-path="peak-calling-and-motif-analysis.html"><a href="peak-calling-and-motif-analysis.html#linkage-plot"><i class="fa fa-check"></i><b>7.1</b> Linkage plot</a></li>
<li class="chapter" data-level="7.2" data-path="peak-calling-and-motif-analysis.html"><a href="peak-calling-and-motif-analysis.html#call-peaks-using-macs2"><i class="fa fa-check"></i><b>7.2</b> Call peaks using MACS2</a></li>
<li class="chapter" data-level="7.3" data-path="peak-calling-and-motif-analysis.html"><a href="peak-calling-and-motif-analysis.html#motif-analysis"><i class="fa fa-check"></i><b>7.3</b> Motif analysis</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="data-integration.html"><a href="data-integration.html"><i class="fa fa-check"></i><b>8</b> Data integration</a></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Hands-on Analysis on Single Cell Multiome</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="dimension-reduction-and-clustering" class="section level1 hasAnchor" number="5">
<h1><span class="header-section-number">Chapter 5</span> Dimension reduction and clustering<a href="dimension-reduction-and-clustering.html#dimension-reduction-and-clustering" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="identify-top-variable-features" class="section level2 hasAnchor" number="5.1">
<h2><span class="header-section-number">5.1</span> Identify top variable features<a href="dimension-reduction-and-clustering.html#identify-top-variable-features" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="dimension-reduction-and-clustering.html#cb1-1" tabindex="-1"></a><span class="co"># Finding highly variable genes (feature selection), we return 2000 features per dataset by default</span></span>
<span id="cb1-2"><a href="dimension-reduction-and-clustering.html#cb1-2" tabindex="-1"></a>pbmc <span class="ot">&lt;-</span> <span class="fu">FindVariableFeatures</span>(<span class="at">object =</span> pbmc, <span class="at">selection.method =</span><span class="st">&quot;vst&quot;</span>, </span>
<span id="cb1-3"><a href="dimension-reduction-and-clustering.html#cb1-3" tabindex="-1"></a>                                 <span class="at">mean.cutoff =</span> <span class="fu">c</span>(<span class="fl">0.1</span>,<span class="dv">6</span>), <span class="at">dispersion.cutoff =</span> <span class="fu">c</span>(<span class="fl">0.5</span>,<span class="cn">Inf</span>),<span class="at">verbose=</span><span class="cn">TRUE</span>)</span>
<span id="cb1-4"><a href="dimension-reduction-and-clustering.html#cb1-4" tabindex="-1"></a><span class="fu">length</span>(<span class="fu">VariableFeatures</span>(pbmc))</span>
<span id="cb1-5"><a href="dimension-reduction-and-clustering.html#cb1-5" tabindex="-1"></a><span class="co"># [1] 2000 # return top 2000 variable features by default</span></span></code></pre></div>
<pre><code>##  [1] &quot;JCHAIN&quot;  &quot;PTGDS&quot;   &quot;IGKC&quot;    &quot;IGHA1&quot;   &quot;TXNDC5&quot;  &quot;TCF4&quot;    &quot;SOX5&quot;   
##  [8] &quot;MZB1&quot;    &quot;ZNF385D&quot; &quot;CDKN1C&quot;</code></pre>
<pre><code>## When using repel, set xnudge and ynudge to 0 for optimal results</code></pre>
<pre><code>## Warning in scale_x_log10(): log-10 transformation introduced infinite values.
## log-10 transformation introduced infinite values.</code></pre>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-4"></span>
<img src="bookdown-demo_files/figure-html/unnamed-chunk-4-1.png" alt="Label the top 10 variable features" width="100%" />
<p class="caption">
Figure 5.1: Label the top 10 variable features
</p>
</div>
</div>
<div id="dimension-reduction" class="section level2 hasAnchor" number="5.2">
<h2><span class="header-section-number">5.2</span> Dimension reduction<a href="dimension-reduction-and-clustering.html#dimension-reduction" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Run separately for RNA and ATAC. <code>min.dist</code> and <code>spread</code> controls how tightly the points are clustered.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="dimension-reduction-and-clustering.html#cb5-1" tabindex="-1"></a><span class="co"># RNA analysis</span></span>
<span id="cb5-2"><a href="dimension-reduction-and-clustering.html#cb5-2" tabindex="-1"></a><span class="fu">DefaultAssay</span>(pbmc) <span class="ot">&lt;-</span> <span class="st">&quot;RNA&quot;</span></span>
<span id="cb5-3"><a href="dimension-reduction-and-clustering.html#cb5-3" tabindex="-1"></a>pbmc <span class="ot">&lt;-</span> <span class="fu">RunPCA</span>(pbmc) </span>
<span id="cb5-4"><a href="dimension-reduction-and-clustering.html#cb5-4" tabindex="-1"></a><span class="fu">DimPlot</span>(<span class="at">object =</span> pbmc)</span></code></pre></div>
<p><img src="bookdown-demo_files/figure-html/unnamed-chunk-5-1.png" width="672" /></p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="dimension-reduction-and-clustering.html#cb6-1" tabindex="-1"></a><span class="co"># determin if cell cycle has major variation</span></span>
<span id="cb6-2"><a href="dimension-reduction-and-clustering.html#cb6-2" tabindex="-1"></a><span class="co"># Running a PCA on cell cycle genes reveals</span></span>
<span id="cb6-3"><a href="dimension-reduction-and-clustering.html#cb6-3" tabindex="-1"></a><span class="fu">DimPlot</span>(pbmc,<span class="at">reduction =</span> <span class="st">&quot;pca&quot;</span>,<span class="at">group.by=</span> <span class="st">&quot;Phase&quot;</span>,<span class="at">split.by =</span> <span class="st">&quot;Phase&quot;</span>)</span></code></pre></div>
<p><img src="bookdown-demo_files/figure-html/unnamed-chunk-5-2.png" width="672" /></p>
<p>Other visualization figures for determining dimensionality:</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="dimension-reduction-and-clustering.html#cb7-1" tabindex="-1"></a><span class="co"># visualize the top genes associated with reduction components for the first 2 PCs</span></span>
<span id="cb7-2"><a href="dimension-reduction-and-clustering.html#cb7-2" tabindex="-1"></a><span class="fu">VizDimLoadings</span>(<span class="at">object =</span> pbmc, <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-6"></span>
<img src="bookdown-demo_files/figure-html/unnamed-chunk-6-1.png" alt="Variations in top genes for the first two PCs" width="100%" />
<p class="caption">
Figure 5.2: Variations in top genes for the first two PCs
</p>
</div>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="dimension-reduction-and-clustering.html#cb8-1" tabindex="-1"></a><span class="co"># use heatmap</span></span>
<span id="cb8-2"><a href="dimension-reduction-and-clustering.html#cb8-2" tabindex="-1"></a><span class="fu">DimHeatmap</span>(<span class="at">object =</span> pbmc, <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">9</span>, <span class="at">cells =</span> <span class="dv">500</span>, <span class="at">balanced =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-7"></span>
<img src="bookdown-demo_files/figure-html/unnamed-chunk-7-1.png" alt="Heatmap for visualizing heterogeneity in PCs" width="100%" />
<p class="caption">
Figure 5.3: Heatmap for visualizing heterogeneity in PCs
</p>
</div>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="dimension-reduction-and-clustering.html#cb9-1" tabindex="-1"></a><span class="fu">ElbowPlot</span>(<span class="at">object =</span> pbmc)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-8"></span>
<img src="bookdown-demo_files/figure-html/unnamed-chunk-8-1.png" alt="Checking elbow plot is the most common way for determining dimensionality" width="100%" />
<p class="caption">
Figure 5.4: Checking elbow plot is the most common way for determining dimensionality
</p>
</div>
<p>We will do dimension reduction for scATAC-seq as well.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="dimension-reduction-and-clustering.html#cb10-1" tabindex="-1"></a><span class="co"># ATAC analysis</span></span>
<span id="cb10-2"><a href="dimension-reduction-and-clustering.html#cb10-2" tabindex="-1"></a><span class="fu">DefaultAssay</span>(pbmc) <span class="ot">&lt;-</span> <span class="st">&quot;ATAC&quot;</span></span>
<span id="cb10-3"><a href="dimension-reduction-and-clustering.html#cb10-3" tabindex="-1"></a>pbmc <span class="ot">&lt;-</span> <span class="fu">RunTFIDF</span>(pbmc)</span>
<span id="cb10-4"><a href="dimension-reduction-and-clustering.html#cb10-4" tabindex="-1"></a>pbmc <span class="ot">&lt;-</span> <span class="fu">FindTopFeatures</span>(pbmc, <span class="at">min.cutoff =</span> <span class="st">&#39;q0&#39;</span>)</span>
<span id="cb10-5"><a href="dimension-reduction-and-clustering.html#cb10-5" tabindex="-1"></a>pbmc <span class="ot">&lt;-</span> <span class="fu">RunSVD</span>(pbmc)</span>
<span id="cb10-6"><a href="dimension-reduction-and-clustering.html#cb10-6" tabindex="-1"></a>pbmc <span class="ot">&lt;-</span> <span class="fu">RunUMAP</span>(pbmc, <span class="at">reduction =</span> <span class="st">&#39;lsi&#39;</span>, <span class="at">dims =</span> <span class="dv">2</span><span class="sc">:</span><span class="dv">50</span>, <span class="at">reduction.name =</span> <span class="st">&quot;umap.atac&quot;</span>, <span class="at">reduction.key =</span> <span class="st">&quot;atacUMAP_&quot;</span>)</span></code></pre></div>
</div>
<div id="clustering-and-joint-clustering" class="section level2 hasAnchor" number="5.3">
<h2><span class="header-section-number">5.3</span> Clustering and joint clustering<a href="dimension-reduction-and-clustering.html#clustering-and-joint-clustering" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Clustering based on RNA assay. Play around with different resolutions.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="dimension-reduction-and-clustering.html#cb11-1" tabindex="-1"></a><span class="fu">DefaultAssay</span>(pbmc) <span class="ot">&lt;-</span> <span class="st">&quot;RNA&quot;</span></span>
<span id="cb11-2"><a href="dimension-reduction-and-clustering.html#cb11-2" tabindex="-1"></a>pbmc <span class="ot">&lt;-</span> <span class="fu">FindNeighbors</span>(pbmc, <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">50</span>,<span class="at">force.recalc =</span> T)</span>
<span id="cb11-3"><a href="dimension-reduction-and-clustering.html#cb11-3" tabindex="-1"></a>pbmc <span class="ot">&lt;-</span> <span class="fu">FindClusters</span>(pbmc, <span class="at">algorithm =</span> <span class="dv">3</span>, <span class="at">resolution =</span> <span class="fl">0.7</span>, <span class="at">group.singletons=</span>T)</span>
<span id="cb11-4"><a href="dimension-reduction-and-clustering.html#cb11-4" tabindex="-1"></a>pbmc<span class="ot">&lt;-</span><span class="fu">RunUMAP</span>(<span class="at">object =</span> pbmc, <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">50</span>, <span class="at">reduction.name =</span> <span class="st">&quot;umap.rna&quot;</span>, <span class="at">reduction.key =</span> <span class="st">&#39;rnaUMAP_&#39;</span>,</span>
<span id="cb11-5"><a href="dimension-reduction-and-clustering.html#cb11-5" tabindex="-1"></a>                    <span class="at">min.dist=</span><span class="fl">0.3</span>, <span class="at">spread=</span><span class="fl">0.5</span>,<span class="at">assay=</span><span class="st">&quot;SCT&quot;</span>,<span class="at">slot=</span><span class="st">&quot;scale.data&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="dimension-reduction-and-clustering.html#cb12-1" tabindex="-1"></a><span class="fu">DimPlot</span>(pbmc, <span class="at">label =</span> <span class="cn">TRUE</span>, <span class="at">repel =</span> <span class="cn">TRUE</span>, <span class="at">reduction =</span> <span class="st">&quot;umap.rna&quot;</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-12"></span>
<img src="bookdown-demo_files/figure-html/unnamed-chunk-12-1.png" alt="Resolution=0.7" width="100%" />
<p class="caption">
Figure 5.5: Resolution=0.7
</p>
</div>
<p>Generate a resolution tree for res=0.1 to 1.5 to select a suitable resolution.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="dimension-reduction-and-clustering.html#cb13-1" tabindex="-1"></a><span class="fu">DefaultAssay</span>(pbmc) <span class="ot">&lt;-</span> <span class="st">&quot;RNA&quot;</span></span>
<span id="cb13-2"><a href="dimension-reduction-and-clustering.html#cb13-2" tabindex="-1"></a>resolutions <span class="ot">&lt;-</span> <span class="fu">seq</span>(<span class="fl">0.1</span>, <span class="fl">1.5</span>, <span class="at">by =</span> <span class="fl">0.1</span>)</span>
<span id="cb13-3"><a href="dimension-reduction-and-clustering.html#cb13-3" tabindex="-1"></a><span class="co"># Initialize a list to store the clustering results</span></span>
<span id="cb13-4"><a href="dimension-reduction-and-clustering.html#cb13-4" tabindex="-1"></a>clustering.list <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb13-5"><a href="dimension-reduction-and-clustering.html#cb13-5" tabindex="-1"></a><span class="co"># Loop over the resolution values and perform clustering</span></span>
<span id="cb13-6"><a href="dimension-reduction-and-clustering.html#cb13-6" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(resolutions)){</span>
<span id="cb13-7"><a href="dimension-reduction-and-clustering.html#cb13-7" tabindex="-1"></a>    <span class="co"># Run the FindClusters function with the resolution parameter</span></span>
<span id="cb13-8"><a href="dimension-reduction-and-clustering.html#cb13-8" tabindex="-1"></a>    pbmc <span class="ot">&lt;-</span> <span class="fu">FindClusters</span>(pbmc, <span class="at">algorithm =</span> <span class="dv">3</span>, <span class="at">resolution =</span> resolutions[i], <span class="at">group.singletons=</span>T, <span class="at">verbose =</span> <span class="cn">TRUE</span>)</span>
<span id="cb13-9"><a href="dimension-reduction-and-clustering.html#cb13-9" tabindex="-1"></a>    <span class="co"># Store the cluster labels in the list</span></span>
<span id="cb13-10"><a href="dimension-reduction-and-clustering.html#cb13-10" tabindex="-1"></a>    clustering.list[[i]] <span class="ot">&lt;-</span> <span class="fu">Idents</span>(pbmc) </span>
<span id="cb13-11"><a href="dimension-reduction-and-clustering.html#cb13-11" tabindex="-1"></a>    <span class="fu">print</span>(i)</span>
<span id="cb13-12"><a href="dimension-reduction-and-clustering.html#cb13-12" tabindex="-1"></a>}</span></code></pre></div>
<pre><code>## Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck
## 
## Number of nodes: 10413
## Number of edges: 497123
## 
## Running smart local moving algorithm...
## Maximum modularity in 10 random starts: 0.9643
## Number of communities: 8
## Elapsed time: 8 seconds
## [1] 1
## Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck
## 
## Number of nodes: 10413
## Number of edges: 497123
## 
## Running smart local moving algorithm...
## Maximum modularity in 10 random starts: 0.9444
## Number of communities: 10
## Elapsed time: 7 seconds
## [1] 2
## Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck
## 
## Number of nodes: 10413
## Number of edges: 497123
## 
## Running smart local moving algorithm...
## Maximum modularity in 10 random starts: 0.9276
## Number of communities: 13
## Elapsed time: 6 seconds
## [1] 3
## Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck
## 
## Number of nodes: 10413
## Number of edges: 497123
## 
## Running smart local moving algorithm...
## Maximum modularity in 10 random starts: 0.9140
## Number of communities: 13
## Elapsed time: 6 seconds
## [1] 4
## Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck
## 
## Number of nodes: 10413
## Number of edges: 497123
## 
## Running smart local moving algorithm...
## Maximum modularity in 10 random starts: 0.9004
## Number of communities: 14
## Elapsed time: 6 seconds
## [1] 5
## Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck
## 
## Number of nodes: 10413
## Number of edges: 497123
## 
## Running smart local moving algorithm...
## Maximum modularity in 10 random starts: 0.8869
## Number of communities: 14
## Elapsed time: 6 seconds
## [1] 6
## Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck
## 
## Number of nodes: 10413
## Number of edges: 497123
## 
## Running smart local moving algorithm...
## Maximum modularity in 10 random starts: 0.8734
## Number of communities: 15
## Elapsed time: 6 seconds
## [1] 7
## Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck
## 
## Number of nodes: 10413
## Number of edges: 497123
## 
## Running smart local moving algorithm...
## Maximum modularity in 10 random starts: 0.8618
## Number of communities: 17
## Elapsed time: 6 seconds
## [1] 8
## Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck
## 
## Number of nodes: 10413
## Number of edges: 497123
## 
## Running smart local moving algorithm...
## Maximum modularity in 10 random starts: 0.8514
## Number of communities: 17
## Elapsed time: 6 seconds
## [1] 9
## Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck
## 
## Number of nodes: 10413
## Number of edges: 497123
## 
## Running smart local moving algorithm...
## Maximum modularity in 10 random starts: 0.8411
## Number of communities: 17
## Elapsed time: 6 seconds
## [1] 10
## Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck
## 
## Number of nodes: 10413
## Number of edges: 497123
## 
## Running smart local moving algorithm...
## Maximum modularity in 10 random starts: 0.8316
## Number of communities: 18
## Elapsed time: 6 seconds
## [1] 11
## Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck
## 
## Number of nodes: 10413
## Number of edges: 497123
## 
## Running smart local moving algorithm...
## Maximum modularity in 10 random starts: 0.8225
## Number of communities: 19
## Elapsed time: 5 seconds
## [1] 12
## Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck
## 
## Number of nodes: 10413
## Number of edges: 497123
## 
## Running smart local moving algorithm...
## Maximum modularity in 10 random starts: 0.8141
## Number of communities: 20
## Elapsed time: 5 seconds
## [1] 13
## Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck
## 
## Number of nodes: 10413
## Number of edges: 497123
## 
## Running smart local moving algorithm...
## Maximum modularity in 10 random starts: 0.8064
## Number of communities: 20
## Elapsed time: 5 seconds
## [1] 14
## Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck
## 
## Number of nodes: 10413
## Number of edges: 497123
## 
## Running smart local moving algorithm...
## Maximum modularity in 10 random starts: 0.7984
## Number of communities: 21
## Elapsed time: 5 seconds
## [1] 15</code></pre>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="dimension-reduction-and-clustering.html#cb15-1" tabindex="-1"></a><span class="fu">names</span>(clustering.list)<span class="ot">&lt;-</span><span class="fu">paste0</span>(<span class="st">&quot;res_&quot;</span>,resolutions)</span></code></pre></div>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="dimension-reduction-and-clustering.html#cb16-1" tabindex="-1"></a><span class="co"># Load the clustree package</span></span>
<span id="cb16-2"><a href="dimension-reduction-and-clustering.html#cb16-2" tabindex="-1"></a><span class="fu">library</span>(clustree)</span>
<span id="cb16-3"><a href="dimension-reduction-and-clustering.html#cb16-3" tabindex="-1"></a>df<span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">lapply</span>(clustering.list,as.factor))</span>
<span id="cb16-4"><a href="dimension-reduction-and-clustering.html#cb16-4" tabindex="-1"></a><span class="co"># Plot the cluster tree for different resolution values</span></span>
<span id="cb16-5"><a href="dimension-reduction-and-clustering.html#cb16-5" tabindex="-1"></a><span class="fu">clustree</span>(df, <span class="at">prefix =</span> <span class="st">&quot;res_&quot;</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-14"></span>
<img src="bookdown-demo_files/figure-html/unnamed-chunk-14-1.png" alt="Clusters based on RNA assay vs joint assays" width="100%" />
<p class="caption">
Figure 5.6: Clusters based on RNA assay vs joint assays
</p>
</div>
<p>Compare: resolution=0.1, resolution=0.6 and resolution=1.1.
s</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="dimension-reduction-and-clustering.html#cb17-1" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> <span class="fu">DimPlot</span>(pbmc, <span class="at">reduction =</span> <span class="st">&quot;wnn.umap&quot;</span>, <span class="at">group.by =</span> <span class="st">&quot;RNA_snn_res.0.1&quot;</span>, <span class="at">label =</span> <span class="cn">TRUE</span>, <span class="at">label.size =</span> <span class="fl">2.5</span>, <span class="at">repel =</span> <span class="cn">TRUE</span>) <span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="st">&quot;RNA clusters res=0.1&quot;</span>)</span>
<span id="cb17-2"><a href="dimension-reduction-and-clustering.html#cb17-2" tabindex="-1"></a>p2 <span class="ot">&lt;-</span> <span class="fu">DimPlot</span>(pbmc, <span class="at">reduction =</span> <span class="st">&quot;wnn.umap&quot;</span>, <span class="at">group.by =</span> <span class="st">&quot;RNA_snn_res.0.7&quot;</span>, <span class="at">label =</span> <span class="cn">TRUE</span>, <span class="at">label.size =</span> <span class="fl">2.5</span>, <span class="at">repel =</span> <span class="cn">TRUE</span>) <span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="st">&quot;RNA clusters res=0.7&quot;</span>)</span>
<span id="cb17-3"><a href="dimension-reduction-and-clustering.html#cb17-3" tabindex="-1"></a>p3 <span class="ot">&lt;-</span> <span class="fu">DimPlot</span>(pbmc, <span class="at">reduction =</span> <span class="st">&quot;wnn.umap&quot;</span>, <span class="at">group.by =</span> <span class="st">&quot;RNA_snn_res.1.1&quot;</span>, <span class="at">label =</span> <span class="cn">TRUE</span>, <span class="at">label.size =</span> <span class="fl">2.5</span>, <span class="at">repel =</span> <span class="cn">TRUE</span>) <span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="st">&quot;RNA clusters res=1.1&quot;</span>)</span>
<span id="cb17-4"><a href="dimension-reduction-and-clustering.html#cb17-4" tabindex="-1"></a>p1 <span class="sc">+</span> p2 <span class="sc">+</span> p3 <span class="sc">&amp;</span> <span class="fu">NoLegend</span>() <span class="sc">&amp;</span> <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="fl">0.5</span>))</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-15"></span>
<img src="bookdown-demo_files/figure-html/unnamed-chunk-15-1.png" alt="Clusters based on RNA assay vs joint assays" width="250%" />
<p class="caption">
Figure 5.7: Clusters based on RNA assay vs joint assays
</p>
</div>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="dimension-reduction-and-clustering.html#cb18-1" tabindex="-1"></a><span class="co"># We exclude the first dimension for ATAC as this is typically correlated with sequencing depth</span></span>
<span id="cb18-2"><a href="dimension-reduction-and-clustering.html#cb18-2" tabindex="-1"></a>pbmc <span class="ot">&lt;-</span> <span class="fu">FindMultiModalNeighbors</span>(pbmc, <span class="at">reduction.list =</span> <span class="fu">list</span>(<span class="st">&quot;pca&quot;</span>, <span class="st">&quot;lsi&quot;</span>), <span class="at">dims.list =</span> <span class="fu">list</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">50</span>, <span class="dv">2</span><span class="sc">:</span><span class="dv">50</span>))</span>
<span id="cb18-3"><a href="dimension-reduction-and-clustering.html#cb18-3" tabindex="-1"></a>pbmc <span class="ot">&lt;-</span> <span class="fu">RunUMAP</span>(pbmc, <span class="at">nn.name =</span> <span class="st">&quot;weighted.nn&quot;</span>, <span class="at">reduction.name =</span> <span class="st">&quot;wnn.umap&quot;</span>, <span class="at">reduction.key =</span> <span class="st">&quot;wnnUMAP_&quot;</span>,<span class="at">min.dist=</span><span class="fl">0.3</span>, <span class="at">spread=</span><span class="fl">0.5</span>)</span>
<span id="cb18-4"><a href="dimension-reduction-and-clustering.html#cb18-4" tabindex="-1"></a>pbmc <span class="ot">&lt;-</span> <span class="fu">FindClusters</span>(pbmc, <span class="at">graph.name =</span> <span class="st">&quot;wsnn&quot;</span>, <span class="at">algorithm =</span> <span class="dv">3</span>, <span class="at">resolution =</span> <span class="fl">0.6</span>, <span class="at">group.singletons=</span>T,<span class="at">verbose =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<p>You can further generate sub-clusters within large cluster to explore underlying structure</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="dimension-reduction-and-clustering.html#cb19-1" tabindex="-1"></a><span class="co"># pbmc &lt;- FindSubCluster(pbmc, cluster = 6, graph.name = &quot;wsnn&quot;, algorithm = 3)</span></span>
<span id="cb19-2"><a href="dimension-reduction-and-clustering.html#cb19-2" tabindex="-1"></a><span class="co"># Idents(pbmc) &lt;- &quot;sub.cluster&quot;</span></span></code></pre></div>
<p>Compare: resolution=0.7 for RNA and resolution=0.6 for joint assays have similar number of clusters.
s</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="dimension-reduction-and-clustering.html#cb20-1" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> <span class="fu">DimPlot</span>(pbmc, <span class="at">reduction =</span> <span class="st">&quot;wnn.umap&quot;</span>, <span class="at">group.by =</span> <span class="st">&quot;RNA_snn_res.0.7&quot;</span>, <span class="at">label =</span> <span class="cn">TRUE</span>, <span class="at">label.size =</span> <span class="fl">2.5</span>, <span class="at">repel =</span> <span class="cn">TRUE</span>) <span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="st">&quot;RNA clusters&quot;</span>)</span>
<span id="cb20-2"><a href="dimension-reduction-and-clustering.html#cb20-2" tabindex="-1"></a>p2 <span class="ot">&lt;-</span> <span class="fu">DimPlot</span>(pbmc, <span class="at">reduction =</span> <span class="st">&quot;wnn.umap&quot;</span>, <span class="at">group.by =</span> <span class="st">&quot;wsnn_res.0.6&quot;</span>, <span class="at">label =</span> <span class="cn">TRUE</span>, <span class="at">label.size =</span> <span class="fl">2.5</span>, <span class="at">repel =</span> <span class="cn">TRUE</span>) <span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="st">&quot;WNN clusters&quot;</span>)</span>
<span id="cb20-3"><a href="dimension-reduction-and-clustering.html#cb20-3" tabindex="-1"></a>p1 <span class="sc">+</span> p2 <span class="sc">&amp;</span> <span class="fu">NoLegend</span>() <span class="sc">&amp;</span> <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="fl">0.5</span>))</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-18"></span>
<img src="bookdown-demo_files/figure-html/unnamed-chunk-18-1.png" alt="Clusters based on RNA assay vs joint assays" width="150%" />
<p class="caption">
Figure 5.8: Clusters based on RNA assay vs joint assays
</p>
</div>
<p>compare: same cluster distribution on different assay space.</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="dimension-reduction-and-clustering.html#cb21-1" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> <span class="fu">DimPlot</span>(pbmc, <span class="at">reduction =</span> <span class="st">&quot;umap.rna&quot;</span>, <span class="at">group.by =</span> <span class="st">&quot;seurat_clusters&quot;</span>, <span class="at">label =</span> <span class="cn">TRUE</span>, <span class="at">label.size =</span> <span class="fl">2.5</span>, <span class="at">repel =</span> <span class="cn">TRUE</span>) <span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="st">&quot;RNA&quot;</span>)</span>
<span id="cb21-2"><a href="dimension-reduction-and-clustering.html#cb21-2" tabindex="-1"></a>p2 <span class="ot">&lt;-</span> <span class="fu">DimPlot</span>(pbmc, <span class="at">reduction =</span> <span class="st">&quot;umap.atac&quot;</span>, <span class="at">group.by =</span> <span class="st">&quot;seurat_clusters&quot;</span>, <span class="at">label =</span> <span class="cn">TRUE</span>, <span class="at">label.size =</span> <span class="fl">2.5</span>, <span class="at">repel =</span> <span class="cn">TRUE</span>) <span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="st">&quot;ATAC&quot;</span>)</span>
<span id="cb21-3"><a href="dimension-reduction-and-clustering.html#cb21-3" tabindex="-1"></a>p3 <span class="ot">&lt;-</span> <span class="fu">DimPlot</span>(pbmc, <span class="at">reduction =</span> <span class="st">&quot;wnn.umap&quot;</span>, <span class="at">group.by =</span> <span class="st">&quot;seurat_clusters&quot;</span>, <span class="at">label =</span> <span class="cn">TRUE</span>, <span class="at">label.size =</span> <span class="fl">2.5</span>, <span class="at">repel =</span> <span class="cn">TRUE</span>) <span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="st">&quot;WNN&quot;</span>)</span>
<span id="cb21-4"><a href="dimension-reduction-and-clustering.html#cb21-4" tabindex="-1"></a>p1 <span class="sc">+</span> p2 <span class="sc">+</span> p3 <span class="sc">&amp;</span> <span class="fu">NoLegend</span>() <span class="sc">&amp;</span> <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="fl">0.5</span>))</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-19"></span>
<img src="bookdown-demo_files/figure-html/unnamed-chunk-19-1.png" alt="Clusters for single assay UMAP space vs joint assay space" width="250%" />
<p class="caption">
Figure 5.9: Clusters for single assay UMAP space vs joint assay space
</p>
</div>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="dimension-reduction-and-clustering.html#cb22-1" tabindex="-1"></a><span class="fu">saveRDS</span>(pbmc,<span class="at">file=</span><span class="st">&quot;pbmc_clust.rds&quot;</span>)</span></code></pre></div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="normalization-and-cell-cycle-correction.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="differential-expression-de-analysis-and-cell-type-annotation.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/04-cluster.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["bookdown-demo.pdf", "bookdown-demo.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
