<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Hands-on Analysis on Single Cell Multiome</title>
  <meta name="description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  <meta name="generator" content="bookdown 0.39 and GitBook 2.6.7" />

  <meta property="og:title" content="Hands-on Analysis on Single Cell Multiome" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Hands-on Analysis on Single Cell Multiome" />
  
  <meta name="twitter:description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  

<meta name="author" content="Danni Liu" />


<meta name="date" content="2024-06-05" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  

<link rel="next" href="cellranger_arc.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Hands-on Analysis on Single Cell Multiome</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Prerequisites</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#about-data"><i class="fa fa-check"></i><b>1.1</b> About Data</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#server-environment-requirements"><i class="fa fa-check"></i><b>1.2</b> Server environment requirements</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#r-libraries"><i class="fa fa-check"></i><b>1.3</b> R libraries</a></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#construct-folder-structure"><i class="fa fa-check"></i><b>1.4</b> Construct folder structure</a></li>
<li class="chapter" data-level="1.5" data-path="index.html"><a href="index.html#downloading-fastq-files"><i class="fa fa-check"></i><b>1.5</b> Downloading FASTQ files</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="cellranger_arc.html"><a href="cellranger_arc.html"><i class="fa fa-check"></i><b>2</b> Cell Ranger ARC for preprocessing sequencing data</a>
<ul>
<li class="chapter" data-level="2.1" data-path="cellranger_arc.html"><a href="cellranger_arc.html#cellranger-arc-mkfastq"><i class="fa fa-check"></i><b>2.1</b> <code>cellranger-arc mkfastq</code></a></li>
<li class="chapter" data-level="2.2" data-path="cellranger_arc.html"><a href="cellranger_arc.html#bcl-naming-convention"><i class="fa fa-check"></i><b>2.2</b> BCL naming convention</a></li>
<li class="chapter" data-level="2.3" data-path="cellranger_arc.html"><a href="cellranger_arc.html#cellranger-arc-count"><i class="fa fa-check"></i><b>2.3</b> <code>cellranger-arc count</code></a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="seuratqc.html"><a href="seuratqc.html"><i class="fa fa-check"></i><b>3</b> Obtain data and perform QC using Seurat</a>
<ul>
<li class="chapter" data-level="3.1" data-path="seuratqc.html"><a href="seuratqc.html#download-data"><i class="fa fa-check"></i><b>3.1</b> Download data</a></li>
<li class="chapter" data-level="3.2" data-path="seuratqc.html"><a href="seuratqc.html#load-data-in-r-and-construct-seurat-object"><i class="fa fa-check"></i><b>3.2</b> Load data in R and construct Seurat object</a></li>
<li class="chapter" data-level="3.3" data-path="seuratqc.html"><a href="seuratqc.html#perform-quality-control-using-both-assays"><i class="fa fa-check"></i><b>3.3</b> Perform quality control using both assays</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="normalization-and-cell-cycle-correction.html"><a href="normalization-and-cell-cycle-correction.html"><i class="fa fa-check"></i><b>4</b> Normalization and cell cycle correction</a>
<ul>
<li class="chapter" data-level="4.1" data-path="normalization-and-cell-cycle-correction.html"><a href="normalization-and-cell-cycle-correction.html#normalization"><i class="fa fa-check"></i><b>4.1</b> Normalization</a></li>
<li class="chapter" data-level="4.2" data-path="normalization-and-cell-cycle-correction.html"><a href="normalization-and-cell-cycle-correction.html#cell-cycle-correction-and-scale-data"><i class="fa fa-check"></i><b>4.2</b> Cell cycle correction and scale data</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="dimension-reduction-and-clustering.html"><a href="dimension-reduction-and-clustering.html"><i class="fa fa-check"></i><b>5</b> Dimension reduction and clustering</a>
<ul>
<li class="chapter" data-level="5.1" data-path="dimension-reduction-and-clustering.html"><a href="dimension-reduction-and-clustering.html#identify-top-variable-features"><i class="fa fa-check"></i><b>5.1</b> Identify top variable features</a></li>
<li class="chapter" data-level="5.2" data-path="dimension-reduction-and-clustering.html"><a href="dimension-reduction-and-clustering.html#dimension-reduction"><i class="fa fa-check"></i><b>5.2</b> Dimension reduction</a></li>
<li class="chapter" data-level="5.3" data-path="dimension-reduction-and-clustering.html"><a href="dimension-reduction-and-clustering.html#clustering-and-joint-clustering"><i class="fa fa-check"></i><b>5.3</b> Clustering and joint clustering</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="differential-expression-de-analysis-and-cell-type-annotation.html"><a href="differential-expression-de-analysis-and-cell-type-annotation.html"><i class="fa fa-check"></i><b>6</b> Differential Expression (DE) analysis and cell type annotation</a>
<ul>
<li class="chapter" data-level="6.1" data-path="differential-expression-de-analysis-and-cell-type-annotation.html"><a href="differential-expression-de-analysis-and-cell-type-annotation.html#de-analysis"><i class="fa fa-check"></i><b>6.1</b> DE analysis</a></li>
<li class="chapter" data-level="6.2" data-path="differential-expression-de-analysis-and-cell-type-annotation.html"><a href="differential-expression-de-analysis-and-cell-type-annotation.html#cell-type-annotation"><i class="fa fa-check"></i><b>6.2</b> Cell type annotation</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="peak-calling-and-motif-analysis.html"><a href="peak-calling-and-motif-analysis.html"><i class="fa fa-check"></i><b>7</b> Peak calling and Motif analysis</a>
<ul>
<li class="chapter" data-level="7.1" data-path="peak-calling-and-motif-analysis.html"><a href="peak-calling-and-motif-analysis.html#linkage-plot"><i class="fa fa-check"></i><b>7.1</b> Linkage plot</a></li>
<li class="chapter" data-level="7.2" data-path="peak-calling-and-motif-analysis.html"><a href="peak-calling-and-motif-analysis.html#call-peaks-using-macs2"><i class="fa fa-check"></i><b>7.2</b> Call peaks using MACS2</a></li>
<li class="chapter" data-level="7.3" data-path="peak-calling-and-motif-analysis.html"><a href="peak-calling-and-motif-analysis.html#motif-analysis"><i class="fa fa-check"></i><b>7.3</b> Motif analysis</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="data-integration.html"><a href="data-integration.html"><i class="fa fa-check"></i><b>8</b> Data integration</a></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Hands-on Analysis on Single Cell Multiome</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="header">
<h1 class="title">Hands-on Analysis on Single Cell Multiome</h1>
<p class="author"><em>Danni Liu</em></p>
<p class="date"><em>2024-06-05</em></p>
</div>
<div id="prerequisites" class="section level1 hasAnchor" number="1">
<h1><span class="header-section-number">Chapter 1</span> Prerequisites<a href="index.html#prerequisites" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="about-data" class="section level2 hasAnchor" number="1.1">
<h2><span class="header-section-number">1.1</span> About Data<a href="index.html#about-data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The data used in this hands-on tutorial is a public single cell ‘multiome’ dataset of cryopreserved human peripheral blood mononuclear cells (PBMC) from 10x Genomics. This dataset is collected from a 25-year-old healthy female donor with around 10,000 cells, and scRNA-seq and scATAC-seq profiles were simultaneously sequenced from the same cells. You can find all information about this data and Cell Ranger ARC output reports from <a href="https://support.10xgenomics.com/single-cell-multiome-atac-gex/datasets/1.0.0/pbmc_granulocyte_sorted_10k?">10x Genomics website</a>.</p>
<p>In this hands-on tutorial, we present a complete workflow for analyzing multiome data starting from sequencing data, with an additional section for data integration of multiomics datasets that are collected from different experiments. We will demonstrate how to preprocess raw sequencing data using Cell Ranger ARC, perform data quality control, cluster and annotate cell types, and identify marker genes using differential expression (DE) analysis. Then we will treat the two profiles of scRNA-seq and scATAC-seq as originating from two separate experiments, and perform data integration. Since they were originally measured in the same cells, this provides a ground truth that we can use to assess the accuracy of integration.</p>
</div>
<div id="server-environment-requirements" class="section level2 hasAnchor" number="1.2">
<h2><span class="header-section-number">1.2</span> Server environment requirements<a href="index.html#server-environment-requirements" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We will use the UCI HPC3 environment for analysis. Please navigate to the <a href="https://hpc3.rcic.uci.edu/biojhub4/hub/login">JypyterHub portal</a>, and start a server under <strong>the second option</strong> with “R4.3.3” and “Spatial Transcriptomics” available in the environment. Please request <strong>4 CPU cores</strong>, with <strong>6Gb of memory</strong> for each core.</p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-1"></span>
<img src="images/hpc3.png" alt="JupyterHub server setting" width="100%" />
<p class="caption">
Figure 1.1: JupyterHub server setting
</p>
</div>
<p>Once the server started, click to open “Rstudio” app.</p>
</div>
<div id="r-libraries" class="section level2 hasAnchor" number="1.3">
<h2><span class="header-section-number">1.3</span> R libraries<a href="index.html#r-libraries" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Please make sure to install the following packages before running the codes. <code>Seurat</code> and <code>SeuratData</code> should have installed in the environment if you are using HPC3 server.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="index.html#cb1-1" tabindex="-1"></a><span class="co"># Enter commands in R (or R studio, if installed)</span></span>
<span id="cb1-2"><a href="index.html#cb1-2" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&#39;Seurat&#39;</span>)</span>
<span id="cb1-3"><a href="index.html#cb1-3" tabindex="-1"></a><span class="fu">library</span>(Seurat)</span>
<span id="cb1-4"><a href="index.html#cb1-4" tabindex="-1"></a></span>
<span id="cb1-5"><a href="index.html#cb1-5" tabindex="-1"></a><span class="co">#install.packages(&quot;devtools&quot;)</span></span>
<span id="cb1-6"><a href="index.html#cb1-6" tabindex="-1"></a><span class="co">#library(devtools)</span></span>
<span id="cb1-7"><a href="index.html#cb1-7" tabindex="-1"></a><span class="co">#devtools::install_github(&#39;satijalab/seurat-data&#39;)</span></span>
<span id="cb1-8"><a href="index.html#cb1-8" tabindex="-1"></a></span>
<span id="cb1-9"><a href="index.html#cb1-9" tabindex="-1"></a><span class="fu">setRepositories</span>(<span class="at">ind=</span><span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>) <span class="co"># needed to automatically install Bioconductor dependencies</span></span>
<span id="cb1-10"><a href="index.html#cb1-10" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;Signac&quot;</span>)</span>
<span id="cb1-11"><a href="index.html#cb1-11" tabindex="-1"></a></span>
<span id="cb1-12"><a href="index.html#cb1-12" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">&quot;remotes&quot;</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>)) {</span>
<span id="cb1-13"><a href="index.html#cb1-13" tabindex="-1"></a>  <span class="fu">install.packages</span>(<span class="st">&quot;remotes&quot;</span>)</span>
<span id="cb1-14"><a href="index.html#cb1-14" tabindex="-1"></a>}</span>
<span id="cb1-15"><a href="index.html#cb1-15" tabindex="-1"></a>remotes<span class="sc">::</span><span class="fu">install_github</span>(<span class="st">&quot;mojaveazure/seurat-disk&quot;</span>)</span>
<span id="cb1-16"><a href="index.html#cb1-16" tabindex="-1"></a></span>
<span id="cb1-17"><a href="index.html#cb1-17" tabindex="-1"></a><span class="co"># Installing genome assembly and gene annotation packages</span></span>
<span id="cb1-18"><a href="index.html#cb1-18" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">require</span>(<span class="st">&quot;BiocManager&quot;</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>))</span>
<span id="cb1-19"><a href="index.html#cb1-19" tabindex="-1"></a>    <span class="fu">install.packages</span>(<span class="st">&quot;BiocManager&quot;</span>)</span>
<span id="cb1-20"><a href="index.html#cb1-20" tabindex="-1"></a>BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="fu">c</span>(<span class="st">&#39;BSgenome.Hsapiens.UCSC.hg38&#39;</span>, <span class="st">&#39;EnsDb.Hsapiens.v86&#39;</span>))</span>
<span id="cb1-21"><a href="index.html#cb1-21" tabindex="-1"></a></span>
<span id="cb1-22"><a href="index.html#cb1-22" tabindex="-1"></a><span class="co"># install other packages needed.</span></span>
<span id="cb1-23"><a href="index.html#cb1-23" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;hdf5r&quot;</span>)</span>
<span id="cb1-24"><a href="index.html#cb1-24" tabindex="-1"></a>BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">&quot;biovizBase&quot;</span>)</span>
<span id="cb1-25"><a href="index.html#cb1-25" tabindex="-1"></a>BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">&quot;glmGamPoi&quot;</span>)</span>
<span id="cb1-26"><a href="index.html#cb1-26" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;dplyr&quot;</span>)</span>
<span id="cb1-27"><a href="index.html#cb1-27" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;clustree&quot;</span>)</span>
<span id="cb1-28"><a href="index.html#cb1-28" tabindex="-1"></a>BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">&quot;TFBSTools&quot;</span>)</span>
<span id="cb1-29"><a href="index.html#cb1-29" tabindex="-1"></a>BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">&quot;JASPAR2020&quot;</span>)</span>
<span id="cb1-30"><a href="index.html#cb1-30" tabindex="-1"></a>BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">&quot;BiocParallel&quot;</span>)</span>
<span id="cb1-31"><a href="index.html#cb1-31" tabindex="-1"></a>BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">&quot;motifmatchr&quot;</span>)</span>
<span id="cb1-32"><a href="index.html#cb1-32" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&#39;ggseqlogo&#39;</span>)</span>
<span id="cb1-33"><a href="index.html#cb1-33" tabindex="-1"></a><span class="co"># install.packages(&quot;VariantAnnotation_1.50.0.tgz&quot;, repos = NULL, type=&quot;source&quot;)</span></span></code></pre></div>
<p>If you want to run future analysis using your own laptop, please make sure you have the latest R and RStudio installed. You can refer to <a href="https://posit.co/download/rstudio-desktop/">RStudio website</a> for installation instructions.</p>
<ul>
<li>Seurat installation: <a href="https://satijalab.org/seurat/articles/install_v5" class="uri">https://satijalab.org/seurat/articles/install_v5</a></li>
<li>Signac installation: <a href="https://stuartlab.org/signac/articles/install" class="uri">https://stuartlab.org/signac/articles/install</a></li>
<li>SeuratData installation: <a href="https://github.com/satijalab/seurat-data" class="uri">https://github.com/satijalab/seurat-data</a></li>
</ul>
</div>
<div id="construct-folder-structure" class="section level2 hasAnchor" number="1.4">
<h2><span class="header-section-number">1.4</span> Construct folder structure<a href="index.html#construct-folder-structure" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="index.html#cb2-1" tabindex="-1"></a>pwd <span class="co"># check the current directory</span></span>
<span id="cb2-2"><a href="index.html#cb2-2" tabindex="-1"></a>mkdir <span class="sc">-</span>p {<span class="sc">~</span><span class="er">/</span>cellranger_arc,<span class="sc">~</span><span class="er">/</span>pbmc_data} <span class="co"># create new folders</span></span>
<span id="cb2-3"><a href="index.html#cb2-3" tabindex="-1"></a>ls <span class="co"># list all files</span></span></code></pre></div>
</div>
<div id="downloading-fastq-files" class="section level2 hasAnchor" number="1.5">
<h2><span class="header-section-number">1.5</span> Downloading FASTQ files<a href="index.html#downloading-fastq-files" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We will practice using one pancreas sample with paired single cell GEX and ATAC from the same set of cells in this hands-on practice. You can download FASTQ files for GEX and ATAC using the following codes.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode bash"><code class="sourceCode bash"></code></pre></div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>

<a href="cellranger_arc.html" class="navigation navigation-next navigation-unique" aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/index.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["bookdown-demo.pdf", "bookdown-demo.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
