<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 7 Peak calling and Motif analysis | Hands-on Analysis on Single Cell Multiome</title>
  <meta name="description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  <meta name="generator" content="bookdown 0.39 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 7 Peak calling and Motif analysis | Hands-on Analysis on Single Cell Multiome" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 7 Peak calling and Motif analysis | Hands-on Analysis on Single Cell Multiome" />
  
  <meta name="twitter:description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  

<meta name="author" content="Danni Liu" />


<meta name="date" content="2024-06-07" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="differential-expression-de-analysis-and-cell-type-annotation.html"/>
<link rel="next" href="data-integration.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Hands-on Analysis on Single Cell Multiome</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Prerequisites</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#about-data"><i class="fa fa-check"></i><b>1.1</b> About Data</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#server-environment-requirements"><i class="fa fa-check"></i><b>1.2</b> Server environment requirements</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#r-libraries"><i class="fa fa-check"></i><b>1.3</b> R libraries</a></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#construct-folder-structure-in-terminal"><i class="fa fa-check"></i><b>1.4</b> Construct folder structure in Terminal</a></li>
<li class="chapter" data-level="1.5" data-path="index.html"><a href="index.html#accessing-pancreas-data"><i class="fa fa-check"></i><b>1.5</b> Accessing pancreas data</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="cellranger_arc.html"><a href="cellranger_arc.html"><i class="fa fa-check"></i><b>2</b> Cell Ranger ARC for preprocessing sequencing data</a>
<ul>
<li class="chapter" data-level="2.1" data-path="cellranger_arc.html"><a href="cellranger_arc.html#cellranger-arc-mkfastq"><i class="fa fa-check"></i><b>2.1</b> <code>cellranger-arc mkfastq</code></a></li>
<li class="chapter" data-level="2.2" data-path="cellranger_arc.html"><a href="cellranger_arc.html#bcl-naming-convention"><i class="fa fa-check"></i><b>2.2</b> BCL naming convention</a></li>
<li class="chapter" data-level="2.3" data-path="cellranger_arc.html"><a href="cellranger_arc.html#cellranger-arc-count"><i class="fa fa-check"></i><b>2.3</b> <code>cellranger-arc count</code></a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="seuratqc.html"><a href="seuratqc.html"><i class="fa fa-check"></i><b>3</b> Obtain data and perform QC using Seurat</a>
<ul>
<li class="chapter" data-level="3.1" data-path="seuratqc.html"><a href="seuratqc.html#download-data"><i class="fa fa-check"></i><b>3.1</b> Download data</a></li>
<li class="chapter" data-level="3.2" data-path="seuratqc.html"><a href="seuratqc.html#load-data-in-r-and-construct-seurat-object"><i class="fa fa-check"></i><b>3.2</b> Load data in R and construct Seurat object</a></li>
<li class="chapter" data-level="3.3" data-path="seuratqc.html"><a href="seuratqc.html#perform-quality-control-using-both-assays"><i class="fa fa-check"></i><b>3.3</b> Perform quality control using both assays</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="normalization-and-cell-cycle-correction.html"><a href="normalization-and-cell-cycle-correction.html"><i class="fa fa-check"></i><b>4</b> Normalization and cell cycle correction</a>
<ul>
<li class="chapter" data-level="4.1" data-path="normalization-and-cell-cycle-correction.html"><a href="normalization-and-cell-cycle-correction.html#normalization"><i class="fa fa-check"></i><b>4.1</b> Normalization</a></li>
<li class="chapter" data-level="4.2" data-path="normalization-and-cell-cycle-correction.html"><a href="normalization-and-cell-cycle-correction.html#cell-cycle-correction-and-scale-data"><i class="fa fa-check"></i><b>4.2</b> Cell cycle correction and scale data</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="dimension-reduction-and-clustering.html"><a href="dimension-reduction-and-clustering.html"><i class="fa fa-check"></i><b>5</b> Dimension reduction and clustering</a>
<ul>
<li class="chapter" data-level="5.1" data-path="dimension-reduction-and-clustering.html"><a href="dimension-reduction-and-clustering.html#identify-top-variable-features"><i class="fa fa-check"></i><b>5.1</b> Identify top variable features</a></li>
<li class="chapter" data-level="5.2" data-path="dimension-reduction-and-clustering.html"><a href="dimension-reduction-and-clustering.html#dimension-reduction"><i class="fa fa-check"></i><b>5.2</b> Dimension reduction</a></li>
<li class="chapter" data-level="5.3" data-path="dimension-reduction-and-clustering.html"><a href="dimension-reduction-and-clustering.html#clustering-and-joint-clustering"><i class="fa fa-check"></i><b>5.3</b> Clustering and joint clustering</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="differential-expression-de-analysis-and-cell-type-annotation.html"><a href="differential-expression-de-analysis-and-cell-type-annotation.html"><i class="fa fa-check"></i><b>6</b> Differential Expression (DE) analysis and cell type annotation</a>
<ul>
<li class="chapter" data-level="6.1" data-path="differential-expression-de-analysis-and-cell-type-annotation.html"><a href="differential-expression-de-analysis-and-cell-type-annotation.html#de-analysis"><i class="fa fa-check"></i><b>6.1</b> DE analysis</a></li>
<li class="chapter" data-level="6.2" data-path="differential-expression-de-analysis-and-cell-type-annotation.html"><a href="differential-expression-de-analysis-and-cell-type-annotation.html#cell-type-annotation"><i class="fa fa-check"></i><b>6.2</b> Cell type annotation</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="peak-calling-and-motif-analysis.html"><a href="peak-calling-and-motif-analysis.html"><i class="fa fa-check"></i><b>7</b> Peak calling and Motif analysis</a>
<ul>
<li class="chapter" data-level="7.1" data-path="peak-calling-and-motif-analysis.html"><a href="peak-calling-and-motif-analysis.html#linkage-plot"><i class="fa fa-check"></i><b>7.1</b> Linkage plot</a></li>
<li class="chapter" data-level="7.2" data-path="peak-calling-and-motif-analysis.html"><a href="peak-calling-and-motif-analysis.html#call-peaks-using-macs2"><i class="fa fa-check"></i><b>7.2</b> Call peaks using MACS2</a></li>
<li class="chapter" data-level="7.3" data-path="peak-calling-and-motif-analysis.html"><a href="peak-calling-and-motif-analysis.html#motif-analysis"><i class="fa fa-check"></i><b>7.3</b> Motif analysis</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="data-integration.html"><a href="data-integration.html"><i class="fa fa-check"></i><b>8</b> Data integration</a>
<ul>
<li class="chapter" data-level="8.1" data-path="data-integration.html"><a href="data-integration.html#identifying-anchors-between-scrna-seq-and-scatac-seq-datasets"><i class="fa fa-check"></i><b>8.1</b> Identifying anchors between scRNA-seq and scATAC-seq datasets</a></li>
<li class="chapter" data-level="8.2" data-path="data-integration.html"><a href="data-integration.html#annotate-scatac-seq-cells-via-label-transfer"><i class="fa fa-check"></i><b>8.2</b> Annotate scATAC-seq cells via label transfer</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Hands-on Analysis on Single Cell Multiome</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="peak-calling-and-motif-analysis" class="section level1 hasAnchor" number="7">
<h1><span class="header-section-number">Chapter 7</span> Peak calling and Motif analysis<a href="peak-calling-and-motif-analysis.html#peak-calling-and-motif-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="linkage-plot" class="section level2 hasAnchor" number="7.1">
<h2><span class="header-section-number">7.1</span> Linkage plot<a href="peak-calling-and-motif-analysis.html#linkage-plot" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="peak-calling-and-motif-analysis.html#cb1-1" tabindex="-1"></a><span class="fu">DefaultAssay</span>(pbmc) <span class="ot">&lt;-</span> <span class="st">&quot;ATAC&quot;</span></span>
<span id="cb1-2"><a href="peak-calling-and-motif-analysis.html#cb1-2" tabindex="-1"></a><span class="co"># first compute the GC content for each peak</span></span>
<span id="cb1-3"><a href="peak-calling-and-motif-analysis.html#cb1-3" tabindex="-1"></a>pbmc <span class="ot">&lt;-</span> <span class="fu">RegionStats</span>(pbmc, <span class="at">genome =</span> BSgenome.Hsapiens.UCSC.hg38)</span>
<span id="cb1-4"><a href="peak-calling-and-motif-analysis.html#cb1-4" tabindex="-1"></a><span class="co"># link peaks to genes</span></span>
<span id="cb1-5"><a href="peak-calling-and-motif-analysis.html#cb1-5" tabindex="-1"></a>pbmc <span class="ot">&lt;-</span> <span class="fu">LinkPeaks</span>(</span>
<span id="cb1-6"><a href="peak-calling-and-motif-analysis.html#cb1-6" tabindex="-1"></a>  <span class="at">object =</span> pbmc,</span>
<span id="cb1-7"><a href="peak-calling-and-motif-analysis.html#cb1-7" tabindex="-1"></a>  <span class="at">peak.assay =</span> <span class="st">&quot;ATAC&quot;</span>,</span>
<span id="cb1-8"><a href="peak-calling-and-motif-analysis.html#cb1-8" tabindex="-1"></a>  <span class="at">expression.assay =</span> <span class="st">&quot;RNA&quot;</span>,</span>
<span id="cb1-9"><a href="peak-calling-and-motif-analysis.html#cb1-9" tabindex="-1"></a>  <span class="at">genes.use =</span> <span class="fu">c</span>(<span class="st">&quot;LYZ&quot;</span>, <span class="st">&quot;MS4A1&quot;</span>)</span>
<span id="cb1-10"><a href="peak-calling-and-motif-analysis.html#cb1-10" tabindex="-1"></a>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-4"></span>
<img src="bookdown-demo_files/figure-html/unnamed-chunk-4-1.png" alt="Linkage plot for gene MS4A1 and LYZ" width="100%" />
<p class="caption">
Figure 7.1: Linkage plot for gene MS4A1 and LYZ
</p>
</div>
</div>
<div id="call-peaks-using-macs2" class="section level2 hasAnchor" number="7.2">
<h2><span class="header-section-number">7.2</span> Call peaks using MACS2<a href="peak-calling-and-motif-analysis.html#call-peaks-using-macs2" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Need to install MACS2 in advance <code>pip install MACS2</code>. Recall peaks by celltype.</p>
<p>Peak calling can be performed using the CallPeaks() function, and can either be done separately for different groups of cells, or performed using data from all the cells. To call peaks on each annotated cell type, we can use the group.by argument:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="peak-calling-and-motif-analysis.html#cb2-1" tabindex="-1"></a><span class="co"># this could take a long time because it runs for each cell type</span></span>
<span id="cb2-2"><a href="peak-calling-and-motif-analysis.html#cb2-2" tabindex="-1"></a>peaks <span class="ot">&lt;-</span> <span class="fu">CallPeaks</span>(</span>
<span id="cb2-3"><a href="peak-calling-and-motif-analysis.html#cb2-3" tabindex="-1"></a>  <span class="at">object =</span> pbmc,</span>
<span id="cb2-4"><a href="peak-calling-and-motif-analysis.html#cb2-4" tabindex="-1"></a>  <span class="at">group.by =</span> <span class="st">&quot;predicted.celltype.l2&quot;</span></span>
<span id="cb2-5"><a href="peak-calling-and-motif-analysis.html#cb2-5" tabindex="-1"></a>)</span></code></pre></div>
</div>
<div id="motif-analysis" class="section level2 hasAnchor" number="7.3">
<h2><span class="header-section-number">7.3</span> Motif analysis<a href="peak-calling-and-motif-analysis.html#motif-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Adding motif information to the Seurat object
To add the DNA sequence motif information required for motif analyses, we can run the AddMotifs() function:</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="peak-calling-and-motif-analysis.html#cb3-1" tabindex="-1"></a><span class="co"># add motif information</span></span>
<span id="cb3-2"><a href="peak-calling-and-motif-analysis.html#cb3-2" tabindex="-1"></a><span class="fu">library</span>(TFBSTools)</span>
<span id="cb3-3"><a href="peak-calling-and-motif-analysis.html#cb3-3" tabindex="-1"></a><span class="fu">library</span>(JASPAR2020)</span>
<span id="cb3-4"><a href="peak-calling-and-motif-analysis.html#cb3-4" tabindex="-1"></a><span class="fu">library</span>(BSgenome.Hsapiens.UCSC.hg38)</span>
<span id="cb3-5"><a href="peak-calling-and-motif-analysis.html#cb3-5" tabindex="-1"></a><span class="fu">library</span>(BiocParallel)</span>
<span id="cb3-6"><a href="peak-calling-and-motif-analysis.html#cb3-6" tabindex="-1"></a><span class="fu">library</span>(motifmatchr)</span>
<span id="cb3-7"><a href="peak-calling-and-motif-analysis.html#cb3-7" tabindex="-1"></a><span class="fu">register</span>(<span class="fu">SerialParam</span>())</span>
<span id="cb3-8"><a href="peak-calling-and-motif-analysis.html#cb3-8" tabindex="-1"></a></span>
<span id="cb3-9"><a href="peak-calling-and-motif-analysis.html#cb3-9" tabindex="-1"></a>pfm <span class="ot">&lt;-</span> <span class="fu">getMatrixSet</span>(</span>
<span id="cb3-10"><a href="peak-calling-and-motif-analysis.html#cb3-10" tabindex="-1"></a>  <span class="at">x =</span> JASPAR2020,</span>
<span id="cb3-11"><a href="peak-calling-and-motif-analysis.html#cb3-11" tabindex="-1"></a>  <span class="at">opts =</span> <span class="fu">list</span>(<span class="at">species =</span> <span class="dv">9606</span>, <span class="at">all_versions =</span> <span class="cn">FALSE</span>)</span>
<span id="cb3-12"><a href="peak-calling-and-motif-analysis.html#cb3-12" tabindex="-1"></a>)</span>
<span id="cb3-13"><a href="peak-calling-and-motif-analysis.html#cb3-13" tabindex="-1"></a></span>
<span id="cb3-14"><a href="peak-calling-and-motif-analysis.html#cb3-14" tabindex="-1"></a>pbmc <span class="ot">&lt;-</span> <span class="fu">AddMotifs</span>(</span>
<span id="cb3-15"><a href="peak-calling-and-motif-analysis.html#cb3-15" tabindex="-1"></a>  <span class="at">object =</span> pbmc,</span>
<span id="cb3-16"><a href="peak-calling-and-motif-analysis.html#cb3-16" tabindex="-1"></a>  <span class="at">genome =</span> BSgenome.Hsapiens.UCSC.hg38,</span>
<span id="cb3-17"><a href="peak-calling-and-motif-analysis.html#cb3-17" tabindex="-1"></a>  <span class="at">pfm =</span> pfm,</span>
<span id="cb3-18"><a href="peak-calling-and-motif-analysis.html#cb3-18" tabindex="-1"></a>  <span class="at">assay =</span> <span class="st">&quot;ATAC&quot;</span>,</span>
<span id="cb3-19"><a href="peak-calling-and-motif-analysis.html#cb3-19" tabindex="-1"></a>  <span class="at">verbose =</span> <span class="cn">TRUE</span></span>
<span id="cb3-20"><a href="peak-calling-and-motif-analysis.html#cb3-20" tabindex="-1"></a>)</span>
<span id="cb3-21"><a href="peak-calling-and-motif-analysis.html#cb3-21" tabindex="-1"></a><span class="co">#pbmc[[&quot;ATAC&quot;]]@motifs</span></span></code></pre></div>
<p>To identify potentially important cell-type-specific regulatory sequences, we can search for DNA motifs that are overrepresented in a set of peaks that are differentially accessible between cell types.</p>
<p>Here, we find differentially accessible peaks between CD8 TEM and B Memory cells. For sparse data (such as scATAC-seq), we find it is often necessary to lower the min.pct threshold in FindMarkers() from the default (0.1, which was designed for scRNA-seq data).</p>
<p>We then perform a hypergeometric test to test the probability of observing the motif at the given frequency by chance, comparing with a background set of peaks matched for GC content.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="peak-calling-and-motif-analysis.html#cb4-1" tabindex="-1"></a>da_peaks <span class="ot">&lt;-</span> <span class="fu">FindMarkers</span>(</span>
<span id="cb4-2"><a href="peak-calling-and-motif-analysis.html#cb4-2" tabindex="-1"></a>  <span class="at">object =</span> pbmc,</span>
<span id="cb4-3"><a href="peak-calling-and-motif-analysis.html#cb4-3" tabindex="-1"></a>  <span class="at">ident.1 =</span> <span class="st">&#39;CD8 TEM&#39;</span>,</span>
<span id="cb4-4"><a href="peak-calling-and-motif-analysis.html#cb4-4" tabindex="-1"></a>  <span class="at">ident.2 =</span> <span class="st">&#39;B memory&#39;</span>,</span>
<span id="cb4-5"><a href="peak-calling-and-motif-analysis.html#cb4-5" tabindex="-1"></a>  <span class="at">only.pos =</span> <span class="cn">TRUE</span>,</span>
<span id="cb4-6"><a href="peak-calling-and-motif-analysis.html#cb4-6" tabindex="-1"></a>  <span class="at">test.use =</span> <span class="st">&#39;LR&#39;</span>,</span>
<span id="cb4-7"><a href="peak-calling-and-motif-analysis.html#cb4-7" tabindex="-1"></a>  <span class="at">min.pct =</span> <span class="fl">0.05</span>,</span>
<span id="cb4-8"><a href="peak-calling-and-motif-analysis.html#cb4-8" tabindex="-1"></a>  <span class="at">latent.vars =</span> <span class="st">&#39;nCount_ATAC&#39;</span></span>
<span id="cb4-9"><a href="peak-calling-and-motif-analysis.html#cb4-9" tabindex="-1"></a>)</span>
<span id="cb4-10"><a href="peak-calling-and-motif-analysis.html#cb4-10" tabindex="-1"></a><span class="co"># get top differentially accessible peaks</span></span>
<span id="cb4-11"><a href="peak-calling-and-motif-analysis.html#cb4-11" tabindex="-1"></a>top.da.peak <span class="ot">&lt;-</span> <span class="fu">rownames</span>(da_peaks[da_peaks<span class="sc">$</span>p_val <span class="sc">&lt;</span> <span class="fl">0.005</span> <span class="sc">&amp;</span> da_peaks<span class="sc">$</span>pct<span class="fl">.1</span> <span class="sc">&gt;</span> <span class="fl">0.2</span>, ])</span>
<span id="cb4-12"><a href="peak-calling-and-motif-analysis.html#cb4-12" tabindex="-1"></a></span>
<span id="cb4-13"><a href="peak-calling-and-motif-analysis.html#cb4-13" tabindex="-1"></a><span class="co"># test enrichment</span></span>
<span id="cb4-14"><a href="peak-calling-and-motif-analysis.html#cb4-14" tabindex="-1"></a>enriched.motifs <span class="ot">&lt;-</span> <span class="fu">FindMotifs</span>(</span>
<span id="cb4-15"><a href="peak-calling-and-motif-analysis.html#cb4-15" tabindex="-1"></a>  <span class="at">object =</span> pbmc,</span>
<span id="cb4-16"><a href="peak-calling-and-motif-analysis.html#cb4-16" tabindex="-1"></a>  <span class="at">features =</span> top.da.peak</span>
<span id="cb4-17"><a href="peak-calling-and-motif-analysis.html#cb4-17" tabindex="-1"></a>)</span></code></pre></div>
<p>We can also plot the position weight matrices for the motifs, so we can visualize the different motif sequences.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="peak-calling-and-motif-analysis.html#cb5-1" tabindex="-1"></a><span class="fu">MotifPlot</span>(</span>
<span id="cb5-2"><a href="peak-calling-and-motif-analysis.html#cb5-2" tabindex="-1"></a>  <span class="at">object =</span> pbmc,</span>
<span id="cb5-3"><a href="peak-calling-and-motif-analysis.html#cb5-3" tabindex="-1"></a>  <span class="at">motifs =</span> <span class="fu">head</span>(<span class="fu">rownames</span>(enriched.motifs))</span>
<span id="cb5-4"><a href="peak-calling-and-motif-analysis.html#cb5-4" tabindex="-1"></a>)</span></code></pre></div>
<pre><code>## Warning: The `&lt;scale&gt;` argument of `guides()` cannot be `FALSE`. Use &quot;none&quot; instead as
## of ggplot2 3.3.4.
## ℹ The deprecated feature was likely used in the ggseqlogo package.
##   Please report the issue at &lt;https://github.com/omarwagih/ggseqlogo/issues&gt;.
## This warning is displayed once every 8 hours.
## Call `lifecycle::last_lifecycle_warnings()` to see where this warning was
## generated.</code></pre>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-10"></span>
<img src="bookdown-demo_files/figure-html/unnamed-chunk-10-1.png" alt="Motifs" width="150%" />
<p class="caption">
Figure 7.2: Motifs
</p>
</div>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="peak-calling-and-motif-analysis.html#cb7-1" tabindex="-1"></a><span class="fu">saveRDS</span>(pbmc,<span class="at">file=</span><span class="st">&quot;pbmc_motif.rds&quot;</span>)</span></code></pre></div>
<hr />

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="differential-expression-de-analysis-and-cell-type-annotation.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="data-integration.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/06-motif.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["bookdown-demo.pdf", "bookdown-demo.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
