<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 3 Obtain data and perform QC using Seurat | Hands-on Analysis on Single Cell Multiome</title>
  <meta name="description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  <meta name="generator" content="bookdown 0.39 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 3 Obtain data and perform QC using Seurat | Hands-on Analysis on Single Cell Multiome" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 3 Obtain data and perform QC using Seurat | Hands-on Analysis on Single Cell Multiome" />
  
  <meta name="twitter:description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  

<meta name="author" content="Danni Liu" />


<meta name="date" content="2024-06-07" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="cellranger_arc.html"/>
<link rel="next" href="normalization-and-cell-cycle-correction.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Hands-on Analysis on Single Cell Multiome</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Prerequisites</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#about-data"><i class="fa fa-check"></i><b>1.1</b> About Data</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#server-environment-requirements"><i class="fa fa-check"></i><b>1.2</b> Server environment requirements</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#r-libraries"><i class="fa fa-check"></i><b>1.3</b> R libraries</a></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#construct-folder-structure-in-terminal"><i class="fa fa-check"></i><b>1.4</b> Construct folder structure in Terminal</a></li>
<li class="chapter" data-level="1.5" data-path="index.html"><a href="index.html#accessing-pancreas-data"><i class="fa fa-check"></i><b>1.5</b> Accessing pancreas data</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="cellranger_arc.html"><a href="cellranger_arc.html"><i class="fa fa-check"></i><b>2</b> Cell Ranger ARC for preprocessing sequencing data</a>
<ul>
<li class="chapter" data-level="2.1" data-path="cellranger_arc.html"><a href="cellranger_arc.html#cellranger-arc-mkfastq"><i class="fa fa-check"></i><b>2.1</b> <code>cellranger-arc mkfastq</code></a></li>
<li class="chapter" data-level="2.2" data-path="cellranger_arc.html"><a href="cellranger_arc.html#bcl-naming-convention"><i class="fa fa-check"></i><b>2.2</b> BCL naming convention</a></li>
<li class="chapter" data-level="2.3" data-path="cellranger_arc.html"><a href="cellranger_arc.html#cellranger-arc-count"><i class="fa fa-check"></i><b>2.3</b> <code>cellranger-arc count</code></a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="seuratqc.html"><a href="seuratqc.html"><i class="fa fa-check"></i><b>3</b> Obtain data and perform QC using Seurat</a>
<ul>
<li class="chapter" data-level="3.1" data-path="seuratqc.html"><a href="seuratqc.html#download-data"><i class="fa fa-check"></i><b>3.1</b> Download data</a></li>
<li class="chapter" data-level="3.2" data-path="seuratqc.html"><a href="seuratqc.html#load-data-in-r-and-construct-seurat-object"><i class="fa fa-check"></i><b>3.2</b> Load data in R and construct Seurat object</a></li>
<li class="chapter" data-level="3.3" data-path="seuratqc.html"><a href="seuratqc.html#perform-quality-control-using-both-assays"><i class="fa fa-check"></i><b>3.3</b> Perform quality control using both assays</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="normalization-and-cell-cycle-correction.html"><a href="normalization-and-cell-cycle-correction.html"><i class="fa fa-check"></i><b>4</b> Normalization and cell cycle correction</a>
<ul>
<li class="chapter" data-level="4.1" data-path="normalization-and-cell-cycle-correction.html"><a href="normalization-and-cell-cycle-correction.html#normalization"><i class="fa fa-check"></i><b>4.1</b> Normalization</a></li>
<li class="chapter" data-level="4.2" data-path="normalization-and-cell-cycle-correction.html"><a href="normalization-and-cell-cycle-correction.html#cell-cycle-correction-and-scale-data"><i class="fa fa-check"></i><b>4.2</b> Cell cycle correction and scale data</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="dimension-reduction-and-clustering.html"><a href="dimension-reduction-and-clustering.html"><i class="fa fa-check"></i><b>5</b> Dimension reduction and clustering</a>
<ul>
<li class="chapter" data-level="5.1" data-path="dimension-reduction-and-clustering.html"><a href="dimension-reduction-and-clustering.html#identify-top-variable-features"><i class="fa fa-check"></i><b>5.1</b> Identify top variable features</a></li>
<li class="chapter" data-level="5.2" data-path="dimension-reduction-and-clustering.html"><a href="dimension-reduction-and-clustering.html#dimension-reduction"><i class="fa fa-check"></i><b>5.2</b> Dimension reduction</a></li>
<li class="chapter" data-level="5.3" data-path="dimension-reduction-and-clustering.html"><a href="dimension-reduction-and-clustering.html#clustering-and-joint-clustering"><i class="fa fa-check"></i><b>5.3</b> Clustering and joint clustering</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="differential-expression-de-analysis-and-cell-type-annotation.html"><a href="differential-expression-de-analysis-and-cell-type-annotation.html"><i class="fa fa-check"></i><b>6</b> Differential Expression (DE) analysis and cell type annotation</a>
<ul>
<li class="chapter" data-level="6.1" data-path="differential-expression-de-analysis-and-cell-type-annotation.html"><a href="differential-expression-de-analysis-and-cell-type-annotation.html#de-analysis"><i class="fa fa-check"></i><b>6.1</b> DE analysis</a></li>
<li class="chapter" data-level="6.2" data-path="differential-expression-de-analysis-and-cell-type-annotation.html"><a href="differential-expression-de-analysis-and-cell-type-annotation.html#cell-type-annotation"><i class="fa fa-check"></i><b>6.2</b> Cell type annotation</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="peak-calling-and-motif-analysis.html"><a href="peak-calling-and-motif-analysis.html"><i class="fa fa-check"></i><b>7</b> Peak calling and Motif analysis</a>
<ul>
<li class="chapter" data-level="7.1" data-path="peak-calling-and-motif-analysis.html"><a href="peak-calling-and-motif-analysis.html#linkage-plot"><i class="fa fa-check"></i><b>7.1</b> Linkage plot</a></li>
<li class="chapter" data-level="7.2" data-path="peak-calling-and-motif-analysis.html"><a href="peak-calling-and-motif-analysis.html#call-peaks-using-macs2"><i class="fa fa-check"></i><b>7.2</b> Call peaks using MACS2</a></li>
<li class="chapter" data-level="7.3" data-path="peak-calling-and-motif-analysis.html"><a href="peak-calling-and-motif-analysis.html#motif-analysis"><i class="fa fa-check"></i><b>7.3</b> Motif analysis</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="data-integration.html"><a href="data-integration.html"><i class="fa fa-check"></i><b>8</b> Data integration</a>
<ul>
<li class="chapter" data-level="8.1" data-path="data-integration.html"><a href="data-integration.html#identifying-anchors-between-scrna-seq-and-scatac-seq-datasets"><i class="fa fa-check"></i><b>8.1</b> Identifying anchors between scRNA-seq and scATAC-seq datasets</a></li>
<li class="chapter" data-level="8.2" data-path="data-integration.html"><a href="data-integration.html#annotate-scatac-seq-cells-via-label-transfer"><i class="fa fa-check"></i><b>8.2</b> Annotate scATAC-seq cells via label transfer</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Hands-on Analysis on Single Cell Multiome</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="seuratqc" class="section level1 hasAnchor" number="3">
<h1><span class="header-section-number">Chapter 3</span> Obtain data and perform QC using Seurat<a href="seuratqc.html#seuratqc" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="download-data" class="section level2 hasAnchor" number="3.1">
<h2><span class="header-section-number">3.1</span> Download data<a href="seuratqc.html#download-data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>To replicate results in this tutorial, you need to download the PBMC data from 10X website.</p>
<p>The PBMC multiome dataset is available from <a href="https://support.10xgenomics.com/single-cell-multiome-atac-gex/datasets/1.0.0/pbmc_granulocyte_sorted_10k">10x genomics</a>. The website contains output files from CelL Ranger ARC. Please go to the above website and make sure you download the following files to your data folder:</p>
<ul>
<li>Filtered feature barcode matrix (HDF5)</li>
<li>ATAC Per fragment information file (TSV.GZ)</li>
<li>ATAC Per fragment information index (TSV.GZ index)</li>
</ul>
<p>In Terminal window:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb1-1"><a href="seuratqc.html#cb1-1" tabindex="-1"></a><span class="fu">mkdir</span> pbmc_data</span>
<span id="cb1-2"><a href="seuratqc.html#cb1-2" tabindex="-1"></a><span class="bu">cd</span> pbmc_data</span>
<span id="cb1-3"><a href="seuratqc.html#cb1-3" tabindex="-1"></a><span class="co"># download data</span></span>
<span id="cb1-4"><a href="seuratqc.html#cb1-4" tabindex="-1"></a><span class="fu">wget</span> https://cf.10xgenomics.com/samples/cell-arc/1.0.0/pbmc_granulocyte_sorted_10k/pbmc_granulocyte_sorted_10k_filtered_feature_bc_matrix.h5</span>
<span id="cb1-5"><a href="seuratqc.html#cb1-5" tabindex="-1"></a><span class="fu">wget</span> https://cf.10xgenomics.com/samples/cell-arc/1.0.0/pbmc_granulocyte_sorted_10k/pbmc_granulocyte_sorted_10k_atac_fragments.tsv.gz</span>
<span id="cb1-6"><a href="seuratqc.html#cb1-6" tabindex="-1"></a><span class="fu">wget</span> https://cf.10xgenomics.com/samples/cell-arc/1.0.0/pbmc_granulocyte_sorted_10k/pbmc_granulocyte_sorted_10k_atac_fragments.tsv.gz.tbi</span>
<span id="cb1-7"><a href="seuratqc.html#cb1-7" tabindex="-1"></a></span>
<span id="cb1-8"><a href="seuratqc.html#cb1-8" tabindex="-1"></a><span class="co"># check files</span></span>
<span id="cb1-9"><a href="seuratqc.html#cb1-9" tabindex="-1"></a><span class="fu">ls</span></span></code></pre></div>
</div>
<div id="load-data-in-r-and-construct-seurat-object" class="section level2 hasAnchor" number="3.2">
<h2><span class="header-section-number">3.2</span> Load data in R and construct Seurat object<a href="seuratqc.html#load-data-in-r-and-construct-seurat-object" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>For the following processing and analysis steps, we are going to use <code>Seurat</code> and its sister package <code>Signac</code>. <code>Seurat</code> is a popular package in R that provides users with well-curated functions and workflows. Seurat was first developed for the clustering of scRNA-seq data, but with continuing updates in the last few years, this package has become a popular tool for QC, analysis, and exploration of scRNA-seq data as well. For more information on <code>Seurat</code>, see the <a href="https://satijalab.org/seurat/">Seurat website</a> from the Satija Lab, which has very nice documentation and links to the Satija Lab <a href="https://satijalab.org/publications">publications</a> as well as detailed <a href="https://satijalab.org/seurat/vignettes.html">tutorials and vignettes</a>.</p>
<p>In this notebook, we will use the most recent version of Seurat, version 5.</p>
<p>We first load all required libraries in this tutorial.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="seuratqc.html#cb2-1" tabindex="-1"></a><span class="fu">library</span>(Seurat)</span>
<span id="cb2-2"><a href="seuratqc.html#cb2-2" tabindex="-1"></a><span class="fu">library</span>(SeuratData)</span>
<span id="cb2-3"><a href="seuratqc.html#cb2-3" tabindex="-1"></a><span class="fu">library</span>(Signac)</span>
<span id="cb2-4"><a href="seuratqc.html#cb2-4" tabindex="-1"></a><span class="fu">library</span>(BSgenome.Hsapiens.UCSC.hg38) </span>
<span id="cb2-5"><a href="seuratqc.html#cb2-5" tabindex="-1"></a><span class="fu">library</span>(EnsDb.Hsapiens.v86)</span>
<span id="cb2-6"><a href="seuratqc.html#cb2-6" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb2-7"><a href="seuratqc.html#cb2-7" tabindex="-1"></a><span class="fu">library</span>(cowplot)</span>
<span id="cb2-8"><a href="seuratqc.html#cb2-8" tabindex="-1"></a><span class="fu">library</span>(hdf5r)</span>
<span id="cb2-9"><a href="seuratqc.html#cb2-9" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb2-10"><a href="seuratqc.html#cb2-10" tabindex="-1"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb2-11"><a href="seuratqc.html#cb2-11" tabindex="-1"></a><span class="fu">library</span>(clustree)</span>
<span id="cb2-12"><a href="seuratqc.html#cb2-12" tabindex="-1"></a><span class="fu">library</span>(TFBSTools)</span>
<span id="cb2-13"><a href="seuratqc.html#cb2-13" tabindex="-1"></a><span class="fu">library</span>(JASPAR2020)</span>
<span id="cb2-14"><a href="seuratqc.html#cb2-14" tabindex="-1"></a><span class="fu">library</span>(BiocParallel)</span>
<span id="cb2-15"><a href="seuratqc.html#cb2-15" tabindex="-1"></a><span class="fu">library</span>(motifmatchr)</span></code></pre></div>
<p>Also set working directory to the folder where you saved your downloaded datasets.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="seuratqc.html#cb3-1" tabindex="-1"></a><span class="co"># check current working directory</span></span>
<span id="cb3-2"><a href="seuratqc.html#cb3-2" tabindex="-1"></a><span class="fu">getwd</span>()</span>
<span id="cb3-3"><a href="seuratqc.html#cb3-3" tabindex="-1"></a></span>
<span id="cb3-4"><a href="seuratqc.html#cb3-4" tabindex="-1"></a><span class="co"># set working directory</span></span>
<span id="cb3-5"><a href="seuratqc.html#cb3-5" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">&quot;/dfs6/pub/your_user_name/pbmc_data&quot;</span>)</span>
<span id="cb3-6"><a href="seuratqc.html#cb3-6" tabindex="-1"></a><span class="fu">getwd</span>() <span class="co"># check again</span></span></code></pre></div>
<p>Next, we will read in the <code>.h5</code> file into R and construct Seurat Object with both scRNA-seq and scATAC-seq assays.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="seuratqc.html#cb4-1" tabindex="-1"></a><span class="co"># the 10x hdf5 file contains both data types. </span></span>
<span id="cb4-2"><a href="seuratqc.html#cb4-2" tabindex="-1"></a>inputdata<span class="fl">.10</span>x <span class="ot">&lt;-</span> <span class="fu">Read10X_h5</span>(<span class="st">&quot;pbmc_granulocyte_sorted_10k_filtered_feature_bc_matrix.h5&quot;</span>)</span>
<span id="cb4-3"><a href="seuratqc.html#cb4-3" tabindex="-1"></a><span class="co"># extract RNA and ATAC data</span></span>
<span id="cb4-4"><a href="seuratqc.html#cb4-4" tabindex="-1"></a>rna_counts <span class="ot">&lt;-</span> inputdata<span class="fl">.10</span>x<span class="sc">$</span><span class="st">`</span><span class="at">Gene Expression</span><span class="st">`</span></span>
<span id="cb4-5"><a href="seuratqc.html#cb4-5" tabindex="-1"></a>atac_counts <span class="ot">&lt;-</span> inputdata<span class="fl">.10</span>x<span class="sc">$</span>Peaks</span>
<span id="cb4-6"><a href="seuratqc.html#cb4-6" tabindex="-1"></a></span>
<span id="cb4-7"><a href="seuratqc.html#cb4-7" tabindex="-1"></a><span class="do">## if reading files from &quot;filtered_feature_bc_matrix&quot;</span></span>
<span id="cb4-8"><a href="seuratqc.html#cb4-8" tabindex="-1"></a><span class="co"># inputdata.10x &lt;- Read10X(data.dir = &quot;filtered_feature_bc_matrix&quot;)</span></span></code></pre></div>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="seuratqc.html#cb5-1" tabindex="-1"></a><span class="co"># Create Seurat object</span></span>
<span id="cb5-2"><a href="seuratqc.html#cb5-2" tabindex="-1"></a>pbmc <span class="ot">&lt;-</span> <span class="fu">CreateSeuratObject</span>(<span class="at">counts =</span> rna_counts,<span class="at">project=</span><span class="st">&quot;pbmc&quot;</span>)</span>
<span id="cb5-3"><a href="seuratqc.html#cb5-3" tabindex="-1"></a></span>
<span id="cb5-4"><a href="seuratqc.html#cb5-4" tabindex="-1"></a><span class="co"># Now add in the ATAC-seq data</span></span>
<span id="cb5-5"><a href="seuratqc.html#cb5-5" tabindex="-1"></a><span class="co"># we&#39;ll only use peaks in standard chromosomes</span></span>
<span id="cb5-6"><a href="seuratqc.html#cb5-6" tabindex="-1"></a>grange.counts <span class="ot">&lt;-</span> <span class="fu">StringToGRanges</span>(<span class="fu">rownames</span>(atac_counts), <span class="at">sep =</span> <span class="fu">c</span>(<span class="st">&quot;:&quot;</span>, <span class="st">&quot;-&quot;</span>))</span>
<span id="cb5-7"><a href="seuratqc.html#cb5-7" tabindex="-1"></a>grange.use <span class="ot">&lt;-</span> <span class="fu">seqnames</span>(grange.counts) <span class="sc">%in%</span> <span class="fu">standardChromosomes</span>(grange.counts)</span>
<span id="cb5-8"><a href="seuratqc.html#cb5-8" tabindex="-1"></a>atac_counts <span class="ot">&lt;-</span> atac_counts[<span class="fu">as.vector</span>(grange.use), ]</span>
<span id="cb5-9"><a href="seuratqc.html#cb5-9" tabindex="-1"></a>annotations <span class="ot">&lt;-</span> <span class="fu">GetGRangesFromEnsDb</span>(<span class="at">ensdb =</span> EnsDb.Hsapiens.v86)</span>
<span id="cb5-10"><a href="seuratqc.html#cb5-10" tabindex="-1"></a><span class="fu">seqlevelsStyle</span>(annotations) <span class="ot">&lt;-</span> <span class="st">&#39;UCSC&#39;</span></span>
<span id="cb5-11"><a href="seuratqc.html#cb5-11" tabindex="-1"></a><span class="fu">genome</span>(annotations) <span class="ot">&lt;-</span> <span class="st">&quot;hg38&quot;</span></span>
<span id="cb5-12"><a href="seuratqc.html#cb5-12" tabindex="-1"></a></span>
<span id="cb5-13"><a href="seuratqc.html#cb5-13" tabindex="-1"></a>frag.file <span class="ot">&lt;-</span> <span class="st">&quot;pbmc_granulocyte_sorted_10k_atac_fragments.tsv.gz&quot;</span></span>
<span id="cb5-14"><a href="seuratqc.html#cb5-14" tabindex="-1"></a>chrom_assay <span class="ot">&lt;-</span> <span class="fu">CreateChromatinAssay</span>(</span>
<span id="cb5-15"><a href="seuratqc.html#cb5-15" tabindex="-1"></a>   <span class="at">counts =</span> atac_counts,</span>
<span id="cb5-16"><a href="seuratqc.html#cb5-16" tabindex="-1"></a>   <span class="at">sep =</span> <span class="fu">c</span>(<span class="st">&quot;:&quot;</span>, <span class="st">&quot;-&quot;</span>),</span>
<span id="cb5-17"><a href="seuratqc.html#cb5-17" tabindex="-1"></a>   <span class="at">genome =</span> <span class="st">&#39;hg38&#39;</span>,</span>
<span id="cb5-18"><a href="seuratqc.html#cb5-18" tabindex="-1"></a>   <span class="at">fragments =</span> frag.file,</span>
<span id="cb5-19"><a href="seuratqc.html#cb5-19" tabindex="-1"></a>   <span class="at">min.cells =</span> <span class="dv">10</span>,</span>
<span id="cb5-20"><a href="seuratqc.html#cb5-20" tabindex="-1"></a>   <span class="at">annotation =</span> annotations</span>
<span id="cb5-21"><a href="seuratqc.html#cb5-21" tabindex="-1"></a> )</span>
<span id="cb5-22"><a href="seuratqc.html#cb5-22" tabindex="-1"></a>pbmc[[<span class="st">&quot;ATAC&quot;</span>]] <span class="ot">&lt;-</span> chrom_assay</span>
<span id="cb5-23"><a href="seuratqc.html#cb5-23" tabindex="-1"></a></span>
<span id="cb5-24"><a href="seuratqc.html#cb5-24" tabindex="-1"></a><span class="co"># check data</span></span>
<span id="cb5-25"><a href="seuratqc.html#cb5-25" tabindex="-1"></a>pbmc</span></code></pre></div>
<p>Add something about seurat object structure
Explore Seurat Objects using <code>@</code> or <code>$</code></p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="seuratqc.html#cb6-1" tabindex="-1"></a><span class="co"># check available assays in data</span></span>
<span id="cb6-2"><a href="seuratqc.html#cb6-2" tabindex="-1"></a>pbmc<span class="sc">@</span>active.assay</span>
<span id="cb6-3"><a href="seuratqc.html#cb6-3" tabindex="-1"></a></span>
<span id="cb6-4"><a href="seuratqc.html#cb6-4" tabindex="-1"></a><span class="co">#check a single assay</span></span>
<span id="cb6-5"><a href="seuratqc.html#cb6-5" tabindex="-1"></a>pbmc[[<span class="st">&quot;RNA&quot;</span>]]</span>
<span id="cb6-6"><a href="seuratqc.html#cb6-6" tabindex="-1"></a>pbmc[[<span class="st">&quot;ATAC&quot;</span>]]</span>
<span id="cb6-7"><a href="seuratqc.html#cb6-7" tabindex="-1"></a></span>
<span id="cb6-8"><a href="seuratqc.html#cb6-8" tabindex="-1"></a><span class="co"># check elements within each assay</span></span>
<span id="cb6-9"><a href="seuratqc.html#cb6-9" tabindex="-1"></a></span>
<span id="cb6-10"><a href="seuratqc.html#cb6-10" tabindex="-1"></a>pbmc[[<span class="st">&quot;RNA&quot;</span>]]<span class="sc">$</span>counts</span>
<span id="cb6-11"><a href="seuratqc.html#cb6-11" tabindex="-1"></a></span>
<span id="cb6-12"><a href="seuratqc.html#cb6-12" tabindex="-1"></a><span class="co"># check metadata</span></span>
<span id="cb6-13"><a href="seuratqc.html#cb6-13" tabindex="-1"></a>pbmc<span class="sc">@</span>meta.data</span></code></pre></div>
<p>Saving data periodically to preserve Seurat results. Output data can be large.</p>
</div>
<div id="perform-quality-control-using-both-assays" class="section level2 hasAnchor" number="3.3">
<h2><span class="header-section-number">3.3</span> Perform quality control using both assays<a href="seuratqc.html#perform-quality-control-using-both-assays" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Using <code>DefaultAssay()</code> to specify the assay you want to work on. <code>PercentageFeatureSet</code> calculates the proportion of transcripts mapping to mitochondrial genes. Large amount of mitochondrial contamination is an indicator of dead or dying cells. Voilin plots help to visualize the distribution of each parameter.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="seuratqc.html#cb7-1" tabindex="-1"></a><span class="co"># calculate percentage of mitochondrial genes in each cell</span></span>
<span id="cb7-2"><a href="seuratqc.html#cb7-2" tabindex="-1"></a><span class="fu">DefaultAssay</span>(pbmc) <span class="ot">&lt;-</span> <span class="st">&quot;RNA&quot;</span></span>
<span id="cb7-3"><a href="seuratqc.html#cb7-3" tabindex="-1"></a>pbmc[[<span class="st">&quot;percent.mt&quot;</span>]] <span class="ot">&lt;-</span> <span class="fu">PercentageFeatureSet</span>(pbmc, <span class="at">pattern =</span> <span class="st">&quot;^MT-&quot;</span>)</span>
<span id="cb7-4"><a href="seuratqc.html#cb7-4" tabindex="-1"></a><span class="fu">DefaultAssay</span>(pbmc) <span class="ot">&lt;-</span> <span class="st">&quot;ATAC&quot;</span></span>
<span id="cb7-5"><a href="seuratqc.html#cb7-5" tabindex="-1"></a>pbmc <span class="ot">&lt;-</span> <span class="fu">NucleosomeSignal</span>(pbmc)</span>
<span id="cb7-6"><a href="seuratqc.html#cb7-6" tabindex="-1"></a>pbmc <span class="ot">&lt;-</span> <span class="fu">TSSEnrichment</span>(pbmc)</span>
<span id="cb7-7"><a href="seuratqc.html#cb7-7" tabindex="-1"></a></span>
<span id="cb7-8"><a href="seuratqc.html#cb7-8" tabindex="-1"></a><span class="fu">DensityScatter</span>(pbmc, <span class="at">x =</span> <span class="st">&#39;nCount_ATAC&#39;</span>, <span class="at">y =</span> <span class="st">&#39;TSS.enrichment&#39;</span>, <span class="at">log_x =</span> <span class="cn">TRUE</span>, <span class="at">quantiles =</span> <span class="cn">TRUE</span>)</span>
<span id="cb7-9"><a href="seuratqc.html#cb7-9" tabindex="-1"></a></span>
<span id="cb7-10"><a href="seuratqc.html#cb7-10" tabindex="-1"></a><span class="co"># generate violin plot</span></span>
<span id="cb7-11"><a href="seuratqc.html#cb7-11" tabindex="-1"></a><span class="fu">VlnPlot</span>(pbmc, <span class="at">features =</span> <span class="fu">c</span>( <span class="st">&quot;nCount_RNA&quot;</span>, <span class="st">&quot;nFeature_RNA&quot;</span>, <span class="st">&quot;percent.mt&quot;</span>), <span class="at">ncol =</span> <span class="dv">3</span>,</span>
<span id="cb7-12"><a href="seuratqc.html#cb7-12" tabindex="-1"></a>  <span class="at">log =</span> <span class="cn">TRUE</span>, <span class="at">pt.size =</span> <span class="dv">0</span>) <span class="sc">+</span> <span class="fu">NoLegend</span>()</span>
<span id="cb7-13"><a href="seuratqc.html#cb7-13" tabindex="-1"></a><span class="fu">VlnPlot</span>(pbmc, <span class="at">features =</span> <span class="fu">c</span>(<span class="st">&quot;nCount_ATAC&quot;</span>, <span class="st">&quot;TSS.enrichment&quot;</span>, <span class="st">&quot;nucleosome_signal&quot;</span>), <span class="at">ncol =</span> <span class="dv">3</span>,</span>
<span id="cb7-14"><a href="seuratqc.html#cb7-14" tabindex="-1"></a>  <span class="at">log =</span> <span class="cn">TRUE</span>, <span class="at">pt.size =</span> <span class="dv">0</span>) <span class="sc">+</span> <span class="fu">NoLegend</span>()</span>
<span id="cb7-15"><a href="seuratqc.html#cb7-15" tabindex="-1"></a><span class="co"># change pt.size to show scatter p</span></span></code></pre></div>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="seuratqc.html#cb8-1" tabindex="-1"></a><span class="co"># show plot</span></span>
<span id="cb8-2"><a href="seuratqc.html#cb8-2" tabindex="-1"></a><span class="fu">VlnPlot</span>(pbmc, <span class="at">features =</span> <span class="fu">c</span>( <span class="st">&quot;nCount_RNA&quot;</span>, <span class="st">&quot;nFeature_RNA&quot;</span>, <span class="st">&quot;percent.mt&quot;</span>), <span class="at">ncol =</span> <span class="dv">3</span>,</span>
<span id="cb8-3"><a href="seuratqc.html#cb8-3" tabindex="-1"></a>  <span class="at">log =</span> <span class="cn">TRUE</span>, <span class="at">pt.size =</span> <span class="dv">0</span>) <span class="sc">+</span> <span class="fu">NoLegend</span>()</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-10-1"></span>
<img src="bookdown-demo_files/figure-html/unnamed-chunk-10-1.png" alt="Voilin plot for QC" width="100%" />
<p class="caption">
Figure 3.1: Voilin plot for QC
</p>
</div>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="seuratqc.html#cb9-1" tabindex="-1"></a><span class="fu">VlnPlot</span>(pbmc, <span class="at">features =</span> <span class="fu">c</span>(<span class="st">&quot;nCount_ATAC&quot;</span>, <span class="st">&quot;TSS.enrichment&quot;</span>, <span class="st">&quot;nucleosome_signal&quot;</span>), <span class="at">ncol =</span> <span class="dv">3</span>,</span>
<span id="cb9-2"><a href="seuratqc.html#cb9-2" tabindex="-1"></a>  <span class="at">log =</span> <span class="cn">TRUE</span>, <span class="at">pt.size =</span> <span class="dv">0</span>) <span class="sc">+</span> <span class="fu">NoLegend</span>()</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-10-2"></span>
<img src="bookdown-demo_files/figure-html/unnamed-chunk-10-2.png" alt="Voilin plot for QC" width="100%" />
<p class="caption">
Figure 3.2: Voilin plot for QC
</p>
</div>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="seuratqc.html#cb10-1" tabindex="-1"></a><span class="co"># Scatter plot for nCount_RNA, percent.mito, nFeature_RNA and filtering based on these variables</span></span>
<span id="cb10-2"><a href="seuratqc.html#cb10-2" tabindex="-1"></a><span class="co"># FeatureScatter is typically used to visualize feature-feature relationships, but can be used</span></span>
<span id="cb10-3"><a href="seuratqc.html#cb10-3" tabindex="-1"></a><span class="co"># for anything calculated by the object, i.e. columns in object metadata, PC scores etc.</span></span>
<span id="cb10-4"><a href="seuratqc.html#cb10-4" tabindex="-1"></a></span>
<span id="cb10-5"><a href="seuratqc.html#cb10-5" tabindex="-1"></a>plot1 <span class="ot">&lt;-</span> <span class="fu">FeatureScatter</span>(pbmc, <span class="at">feature1 =</span> <span class="st">&quot;nCount_RNA&quot;</span>, <span class="at">feature2 =</span> <span class="st">&quot;percent.mt&quot;</span>)</span>
<span id="cb10-6"><a href="seuratqc.html#cb10-6" tabindex="-1"></a>plot2 <span class="ot">&lt;-</span> <span class="fu">FeatureScatter</span>(pbmc, <span class="at">feature1 =</span> <span class="st">&quot;nCount_RNA&quot;</span>, <span class="at">feature2 =</span> <span class="st">&quot;nFeature_RNA&quot;</span>)</span>
<span id="cb10-7"><a href="seuratqc.html#cb10-7" tabindex="-1"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb10-8"><a href="seuratqc.html#cb10-8" tabindex="-1"></a>plot1 <span class="sc">+</span> plot2 <span class="sc">+</span><span class="fu">plot_layout</span>(<span class="at">ncol=</span><span class="dv">2</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-11"></span>
<img src="bookdown-demo_files/figure-html/unnamed-chunk-11-1.png" alt="Voilin plot for QC" width="100%" />
<p class="caption">
Figure 3.3: Voilin plot for QC
</p>
</div>
<p>Perform QC on cells based on the parameters, you can change the paramters in the filtering criteria and see how this can change the number fo cells in the filtered object. <strong>Note that you need to save the filtered object into a name different from the original raw data to avoid overwriting.</strong></p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="seuratqc.html#cb11-1" tabindex="-1"></a><span class="co"># play with different filtering cutoffs to see how number of cells change</span></span>
<span id="cb11-2"><a href="seuratqc.html#cb11-2" tabindex="-1"></a>pbmc_qc <span class="ot">&lt;-</span> <span class="fu">subset</span>(</span>
<span id="cb11-3"><a href="seuratqc.html#cb11-3" tabindex="-1"></a>  <span class="at">x =</span> pbmc,</span>
<span id="cb11-4"><a href="seuratqc.html#cb11-4" tabindex="-1"></a>  <span class="at">subset =</span> nCount_ATAC <span class="sc">&lt;</span> <span class="fl">7e4</span> <span class="sc">&amp;</span></span>
<span id="cb11-5"><a href="seuratqc.html#cb11-5" tabindex="-1"></a>    nCount_ATAC <span class="sc">&gt;</span> <span class="fl">5e3</span> <span class="sc">&amp;</span></span>
<span id="cb11-6"><a href="seuratqc.html#cb11-6" tabindex="-1"></a>    nCount_RNA <span class="sc">&lt;</span> <span class="dv">25000</span> <span class="sc">&amp;</span></span>
<span id="cb11-7"><a href="seuratqc.html#cb11-7" tabindex="-1"></a>    nCount_RNA <span class="sc">&gt;</span> <span class="dv">1000</span> <span class="sc">&amp;</span></span>
<span id="cb11-8"><a href="seuratqc.html#cb11-8" tabindex="-1"></a>    percent.mt <span class="sc">&lt;</span> <span class="dv">25</span> <span class="sc">&amp;</span></span>
<span id="cb11-9"><a href="seuratqc.html#cb11-9" tabindex="-1"></a>    nucleosome_signal <span class="sc">&lt;</span> <span class="dv">2</span> <span class="sc">&amp;</span></span>
<span id="cb11-10"><a href="seuratqc.html#cb11-10" tabindex="-1"></a>    TSS.enrichment <span class="sc">&gt;</span> <span class="dv">1</span></span>
<span id="cb11-11"><a href="seuratqc.html#cb11-11" tabindex="-1"></a>)</span>
<span id="cb11-12"><a href="seuratqc.html#cb11-12" tabindex="-1"></a></span>
<span id="cb11-13"><a href="seuratqc.html#cb11-13" tabindex="-1"></a>pbmc_qc</span></code></pre></div>
<p>Saving data periodically to preserve Seurat results. Output data can be large.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="seuratqc.html#cb12-1" tabindex="-1"></a><span class="fu">saveRDS</span>(pbmc_qc,<span class="at">file=</span><span class="st">&quot;pbmc_qc.rds&quot;</span>)</span></code></pre></div>
<hr />

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="cellranger_arc.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="normalization-and-cell-cycle-correction.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/02-SeuratQC.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["bookdown-demo.pdf", "bookdown-demo.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
